diff --git a/project/plugins.sbt b/project/plugins.sbt
index 4fb8419..8040914 100644
--- a/project/plugins.sbt
+++ b/project/plugins.sbt
@@ -1,4 +1,4 @@
-resolvers += Resolver.url("joprice-sbt-plugins", url("https://dl.bintray.com/content/joprice/sbt-plugins"))(Resolver.ivyStylePatterns)
+resolvers += Resolver.file("sbt-jni", file("$HOME/.ivy2/local/com.github.joprice/sbt-jni"))(Resolver.ivyStylePatterns)

 addSbtPlugin("com.github.joprice" % "sbt-jni" % "0.2.0")
 addSbtPlugin("com.typesafe.sbt" % "sbt-osgi" % "0.9.4")
