diff --git a/cmake/floating_point.cmake b/cmake/floating_point.cmake
index 3ad7e7e8106..a4d52845a04 100644
--- a/cmake/floating_point.cmake
+++ b/cmake/floating_point.cmake
@@ -49,7 +49,11 @@ SET(code "
 )
 
 CMAKE_PUSH_CHECK_STATE()
-STRING_APPEND(CMAKE_REQUIRED_FLAGS " -O3")
+IF(LINUX AND CMAKE_SYSTEM_PROCESSOR STREQUAL "s390x")
+  STRING_APPEND(CMAKE_REQUIRED_FLAGS " -O2")
+ELSE()
+  STRING_APPEND(CMAKE_REQUIRED_FLAGS " -O3")
+ENDIF()
 
 IF(MY_COMPILER_IS_GNU)
   CHECK_C_SOURCE_RUNS("${code}" HAVE_C_FLOATING_POINT_FUSED_MADD)
