diff --git a/storage/ndb/include/portlib/mt-asm.h b/storage/ndb/include/portlib/mt-asm.h
index 418b8d9c711..85e75aee73b 100644
--- a/storage/ndb/include/portlib/mt-asm.h
+++ b/storage/ndb/include/portlib/mt-asm.h
@@ -143,6 +143,13 @@ xcng(volatile unsigned * addr, int val)

   return prev;
 }
+#elif defined(__s390x__)
+#define NDB_HAVE_MB
+#define NDB_HAVE_RMB
+#define NDB_HAVE_WMB
+#define mb()   { asm volatile("" : : : "memory"); }
+#define rmb()  mb()
+#define wmb()  mb()

 #elif defined(__aarch64__)
 #include <atomic>
