diff --git a/run-tests.sh b/run-tests.sh
index 4e722d211..d7dcac03d 100755
--- a/run-tests.sh
+++ b/run-tests.sh
@@ -118,10 +118,10 @@ function check_dependencies()
     fi

     # Check for netstat
-    env netstat --version > /dev/null 2>&1
-    if [ $? -ne 0 ]; then
-        MISSING="$MISSING netstat"
-    fi
+    #env netstat --version > /dev/null 2>&1
+    #if [ $? -ne 0 ]; then
+    #    MISSING="$MISSING netstat"
+    #fi

     # Check for the Perl Test Harness
     env prove --version > /dev/null 2>&1
@@ -155,10 +155,10 @@ function check_dependencies()
     fi

     # Check for netstat
-    env netstat --version > /dev/null 2>&1
-    if [ $? -ne 0 ]; then
-        MISSING="$MISSING netstat"
-    fi
+    #env netstat --version > /dev/null 2>&1
+    #if [ $? -ne 0 ]; then
+    #    MISSING="$MISSING netstat"
+    #fi

     # check for psutil python package
     test `uname -s` == "Darwin" || test `uname -s` == "FreeBSD" && {
