diff --git a/internal/graalvm_bindist.bzl b/internal/graalvm_bindist.bzl
index 0976063..7ec29b6 100644
--- a/internal/graalvm_bindist.bzl
+++ b/internal/graalvm_bindist.bzl
@@ -73,6 +73,7 @@ def _get_platform(ctx, newdist):
         "x86_64": "x64",
         "amd64": "x64",
         "aarch64": "aarch64",
+        "s390x": "s390x",
     }

     # fix: before bazel5, the `arch` property did not exist on `repository_os`, so we need
