diff --git a/Cargo.toml b/Cargo.toml
index aa877a705e..c2517b9097 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -258,18 +258,18 @@ opt-level = 3
 ## until we have upgraded hyper: https://github.com/influxdata/influxdb_iox/issues/9340
 ## see https://github.com/influxdata/arrow-rs/pull/3
 [patch.crates-io]
-arrow = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-array = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-buffer = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-cast = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-csv = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-data = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-ipc = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-json = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-schema = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-select = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-string = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-ord = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-arrow-flight = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
-parquet = { git = "https://github.com/influxdata/arrow-rs.git", rev = "eae176c21b1ef915227294e8a8a201b6f266031a" }
+arrow = { path = "SOURCE_ROOT/arrow-rs/arrow" }
+arrow-array = { path = "SOURCE_ROOT/arrow-rs/arrow-array" }
+arrow-buffer = { path = "SOURCE_ROOT/arrow-rs/arrow-buffer" }
+arrow-cast = { path = "SOURCE_ROOT/arrow-rs/arrow-cast" }
+arrow-csv = { path = "SOURCE_ROOT/arrow-rs/arrow-csv" }
+arrow-data = { path = "SOURCE_ROOT/arrow-rs/arrow-data" }
+arrow-ipc = { path = "SOURCE_ROOT/arrow-rs/arrow-ipc" }
+arrow-json = { path = "SOURCE_ROOT/arrow-rs/arrow-json" }
+arrow-schema = { path = "SOURCE_ROOT/arrow-rs/arrow-schema" }
+arrow-select = { path = "SOURCE_ROOT/arrow-rs/arrow-select" }
+arrow-string = { path = "SOURCE_ROOT/arrow-rs/arrow-string" }
+arrow-ord = { path = "SOURCE_ROOT/arrow-rs/arrow-ord" }
+arrow-flight = { path = "SOURCE_ROOT/arrow-rs/arrow-flight" }
+parquet = { path = "SOURCE_ROOT/arrow-rs/parquet" }
 object_store = { git = "https://github.com/influxdata/arrow-rs.git", rev = "c946cd81fa12e6588a3be33be08e3d8e9a2770e7" }
diff --git a/influxdb3_server/src/lib.rs b/influxdb3_server/src/lib.rs
index 35bb27b4b4..274c620456 100644
--- a/influxdb3_server/src/lib.rs
+++ b/influxdb3_server/src/lib.rs
@@ -587,11 +587,21 @@ mod tests {
 
         assert_eq!(
             batches[0]["time"].to_data().buffers(),
-            &[Buffer::from([123, 0, 0, 0, 0, 0, 0, 0])]
+            &[Buffer::from({
+                #[cfg(target_endian = "little")]
+                {
+                    [123, 0, 0, 0, 0, 0, 0, 0]
+                }
+                #[cfg(target_endian = "big")]
+                {
+                    [0, 0, 0, 0, 0, 0, 0, 123]
+                }
+            })]
         );
+
         assert_eq!(
             batches[0]["val"].to_data().buffers(),
-            &[Buffer::from(1_u64.to_le_bytes())]
+            &[Buffer::from(1_u64.to_ne_bytes())]
         );
 
         shutdown.cancel();
diff --git a/influxdb3_server/src/query_executor/mod.rs b/influxdb3_server/src/query_executor/mod.rs
index 920c07e4d1..c7364708cb 100644
--- a/influxdb3_server/src/query_executor/mod.rs
+++ b/influxdb3_server/src/query_executor/mod.rs
@@ -987,6 +987,9 @@ mod tests {
                     "+------------+------------+-----------+----------+----------+",
                     "| table_name | size_bytes | row_count | min_time | max_time |",
                     "+------------+------------+-----------+----------+----------+",
+                    #[cfg(target_endian = "big")]
+                    "| cpu        | 2058       | 3         | 0        | 20       |",
+                    #[cfg(target_endian = "little")]
                     "| cpu        | 2061       | 3         | 0        | 20       |",
                     "| cpu        | 2061       | 3         | 30       | 50       |",
                     "| cpu        | 2061       | 3         | 60       | 80       |",
@@ -1002,6 +1005,9 @@ mod tests {
                     "+------------+------------+-----------+----------+----------+",
                     "| table_name | size_bytes | row_count | min_time | max_time |",
                     "+------------+------------+-----------+----------+----------+",
+                    #[cfg(target_endian = "big")]
+                    "| mem        | 2058       | 3         | 0        | 20       |",
+                    #[cfg(target_endian = "little")]
                     "| mem        | 2061       | 3         | 0        | 20       |",
                     "| mem        | 2061       | 3         | 30       | 50       |",
                     "| mem        | 2061       | 3         | 60       | 80       |",
@@ -1016,9 +1022,15 @@ mod tests {
                     "+------------+------------+-----------+----------+----------+",
                     "| table_name | size_bytes | row_count | min_time | max_time |",
                     "+------------+------------+-----------+----------+----------+",
+                    #[cfg(target_endian = "big")]
+                    "| cpu        | 2058       | 3         | 0        | 20       |",
+                    #[cfg(target_endian = "little")]
                     "| cpu        | 2061       | 3         | 0        | 20       |",
                     "| cpu        | 2061       | 3         | 30       | 50       |",
                     "| cpu        | 2061       | 3         | 60       | 80       |",
+                    #[cfg(target_endian = "big")]
+                    "| mem        | 2058       | 3         | 0        | 20       |",
+                    #[cfg(target_endian = "little")]
                     "| mem        | 2061       | 3         | 0        | 20       |",
                     "| mem        | 2061       | 3         | 30       | 50       |",
                     "| mem        | 2061       | 3         | 60       | 80       |",
@@ -1034,6 +1046,9 @@ mod tests {
                     "+------------+------------+-----------+----------+----------+",
                     "| table_name | size_bytes | row_count | min_time | max_time |",
                     "+------------+------------+-----------+----------+----------+",
+                    #[cfg(target_endian = "big")]
+                    "| cpu        | 2058       | 3         | 0        | 20       |",
+                    #[cfg(target_endian = "little")]
                     "| cpu        | 2061       | 3         | 0        | 20       |",
                     "| cpu        | 2061       | 3         | 30       | 50       |",
                     "| cpu        | 2061       | 3         | 60       | 80       |",
