--- Lib/test/test_ssl.py.orig	2018-08-20 13:08:52.340780063 -0400
+++ Lib/test/test_ssl.py	2018-08-20 13:08:52.400780063 -0400
@@ -989,6 +989,12 @@
 
 
     def test_load_dh_params(self):
+        filename = u"dhparam.pem"
+        fs_encoding = sys.getfilesystemencoding()
+        try:
+            filename.encode(fs_encoding)
+        except UnicodeEncodeError:
+            self.skipTest("filename %r cannot be encoded to the filesystem encoding %r" % (filename, fs_encoding))
         ctx = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
         ctx.load_dh_params(DHFILE)
         if os.name != 'nt':
