diff --git a/Makefile b/Makefile
index a2f11d6..fb396ec 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 PACKAGE_NAME?=github.com/projectcalico/app-policy
-GO_BUILD_VER?=v0.58
+GO_BUILD_VER?=v0.59
 
 ORGANIZATION=projectcalico
 SEMAPHORE_PROJECT_ID?=$(SEMAPHORE_APP_POLICY_PROJECT_ID)
@@ -158,14 +158,14 @@ proto/felixbackend.pb.go: proto/felixbackend.proto
 		      $(PROTOC_CONTAINER) \
 		      $(PROTOC_IMPORTS) \
 		      proto/*.proto \
-		      --gogofast_out=plugins=grpc:proto
+		      --gogofaster_out=plugins=grpc:proto
 
 proto/healthz.pb.go: proto/healthz.proto
 	$(DOCKER_RUN) -v $(CURDIR):/src:rw \
 		      $(PROTOC_CONTAINER) \
 		      $(PROTOC_IMPORTS) \
 		      proto/*.proto \
-		      --gogofast_out=plugins=grpc:proto
+		      --gogofaster_out=plugins=grpc:proto
 
 
 # Building the image
