diff --git a/orig/curator-5.8.3/setup.cfg b/curator-5.8.3/setup.cfg
index 9f6eef4..af17c20 100644
--- a/orig/curator-5.8.3/setup.cfg
+++ b/curator-5.8.3/setup.cfg
@@ -24,9 +24,9 @@ classifiers =
 install_requires =
     voluptuous>=0.9.3
     elasticsearch==7.1.0
-    urllib3>=1.24.2,<1.26
+    urllib3==1.24.3
     requests>=2.20.0
-    boto3>=1.9.142
+    boto3==1.15.18
     requests_aws4auth>=0.9
     click>=6.7,<7.0
     pyyaml==3.13
@@ -36,9 +36,9 @@ install_requires =
 setup_requires =
     voluptuous>=0.9.3
     elasticsearch==7.1.0
-    urllib3>=1.24.2,<1.26
+    urllib3==1.24.3
     requests>=2.20.0
-    boto3>=1.9.142
+    boto3==1.15.18
     requests_aws4auth>=0.9
     click>=6.7,<7.0
     pyyaml==3.13
