diff --git a/src/dev/register_git_hook/register_git_hook.js b/src/dev/register_git_hook/register_git_hook.js
index 8820327d3a..fb34c924a0 100644
--- a/src/dev/register_git_hook/register_git_hook.js
+++ b/src/dev/register_git_hook/register_git_hook.js
@@ -37,7 +37,7 @@ const writeFileAsync = promisify(writeFile);
 async function getPrecommitGitHookScriptPath(rootPath) {
   // Retrieves the correct location for the .git dir for
   // every git setup (including git worktree)
-  const gitDirPath = (await gitRevParseAsync(['--git-common-dir'])).trim();
+  const gitDirPath = (await gitRevParseAsync(['--git-dir'])).trim();
 
   return resolve(rootPath, gitDirPath, 'hooks/pre-commit');
 }
