diff --git a/cmake/icu.cmake b/cmake/icu.cmake
index f3c1cf06c7c..c0cbd7aea6e 100644
--- a/cmake/icu.cmake
+++ b/cmake/icu.cmake
@@ -127,6 +127,8 @@ SET(BUNDLED_ICU_PATH ${CMAKE_SOURCE_DIR}/extra/icu/${ICU_VERSION_DIR})
 # (Actually, there's an 'e' for EBCDIC version as well.)
 IF(SOLARIS_SPARC)
   SET(ICUDT_DIR "icudt73b")
+ELSEIF(LINUX AND CMAKE_SYSTEM_PROCESSOR STREQUAL "s390x")
+  SET(ICUDT_DIR "icudt73b")
 ELSE()
   SET(ICUDT_DIR "icudt73l")
 ENDIF()
