Submodule c-deps/rocksdb contains modified content
diff --git a/c-deps/rocksdb/CMakeLists.txt b/c-deps/rocksdb/CMakeLists.txt
index f9b0276d6..5d43bb3dd 100644
--- a/c-deps/rocksdb/CMakeLists.txt
+++ b/c-deps/rocksdb/CMakeLists.txt
@@ -224,7 +224,6 @@ else()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /arch:AVX2")
   else()
     if(NOT HAVE_POWER8 AND NOT HAS_ARMV8_CRC)
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
     endif()
   endif()
 endif()
