--- grpc.gemspec.orig	2019-03-12 14:47:32.527125887 +0530
+++ grpc.gemspec	2019-03-12 14:48:06.534119568 +0530
@@ -29,7 +29,7 @@
   s.require_paths = %w( src/ruby/lib src/ruby/bin src/ruby/pb )
   s.platform      = Gem::Platform::RUBY
 
-  s.add_dependency 'google-protobuf', '~> 3.1'
+  s.add_dependency 'google-protobuf', '3.6.1'
   s.add_dependency 'googleapis-common-protos-types', '~> 1.0.0'
 
   s.add_development_dependency 'bundler',            '~> 1.9'
