diff --git a/configure.ac b/configure.ac
index ef07b97..5aed8c1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1369,10 +1369,10 @@ if test "x$cc_werror_format_security" = "xyes"; then
 fi
 CFLAGS="$saved_CFLAGS"

-dnl check for gcc -Werror=implicit-function-declaration
+dnl check for gcc -Wno-error=implicit-function-declaration
 saved_CFLAGS=$CFLAGS
-CFLAGS="-Werror=implicit-function-declaration"
-AC_MSG_CHECKING([whether $CC accepts -Werror=implicit-function-declaration])
+CFLAGS="-Wno-error=implicit-function-declaration"
+AC_MSG_CHECKING([whether $CC accepts -Wno-error=implicit-function-declaration])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM()], [cc_werror_implicit=yes], [cc_werror_implicit=no])
 echo $cc_werror_implicit
 if test "x$cc_werror_implicit" = "xyes"; then
