diff --git a/buildSrc/build.gradle b/buildSrc/build.gradle
index e5611bd99ad..fa390f2f69b 100644
--- a/buildSrc/build.gradle
+++ b/buildSrc/build.gradle
@@ -86,7 +86,7 @@ dependencies {
   api 'com.netflix.nebula:nebula-publishing-plugin:4.4.4'
   api 'com.netflix.nebula:gradle-info-plugin:9.1.1'
   api 'org.apache.rat:apache-rat:0.11'
-  api "org.elasticsearch:jna:5.5.0"
+  api "net.java.dev.jna:jna:5.5.0"
   api 'com.github.jengelman.gradle.plugins:shadow:6.1.0'
   api 'de.thetaphi:forbiddenapis:3.1'
   api 'com.avast.gradle:gradle-docker-compose-plugin:0.13.4'
@@ -171,7 +171,8 @@ if (project != rootProject) {
     distribution project(':distribution:archives:linux-tar')
     distribution project(':distribution:archives:oss-linux-tar')
     distribution project(':distribution:archives:oss-linux-aarch64-tar')
-
+    distribution project(':distribution:archives:linux-s390x-tar')
+    distribution project(':distribution:archives:oss-linux-s390x-tar')
     integTestRuntimeOnly(project(":libs:elasticsearch-core"))
   }
 
diff --git a/buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy b/buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy
index b896e8a8dba..0abb8a3928d 100644
--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy
@@ -191,7 +191,7 @@ class PluginBuildPlugin implements Plugin<Project> {
             compileOnly "org.locationtech.jts:jts-core:${project.versions.jts}"
             compileOnly "org.apache.logging.log4j:log4j-api:${project.versions.log4j}"
             compileOnly "org.apache.logging.log4j:log4j-core:${project.versions.log4j}"
-            compileOnly "org.elasticsearch:jna:${project.versions.jna}"
+            compileOnly "net.java.dev.jna:jna:${project.versions.jna}"
         }
     }
 
diff --git a/buildSrc/src/main/java/org/elasticsearch/gradle/Architecture.java b/buildSrc/src/main/java/org/elasticsearch/gradle/Architecture.java
index c9a7e1eeb8e..dfcdfd0378e 100644
--- a/buildSrc/src/main/java/org/elasticsearch/gradle/Architecture.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/Architecture.java
@@ -11,7 +11,8 @@ package org.elasticsearch.gradle;
 public enum Architecture {
 
     X64,
-    AARCH64;
+    AARCH64,
+    S390X;
 
     public static Architecture current() {
         final String architecture = System.getProperty("os.arch", "");
@@ -19,6 +20,8 @@ public enum Architecture {
             case "amd64":
             case "x86_64":
                 return X64;
+           case "s390x":
+                return S390X;
             case "aarch64":
                 return AARCH64;
             default:
diff --git a/buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java b/buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java
index c9d2f8a0f70..d5f3ab202d6 100644
--- a/buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java
@@ -24,7 +24,7 @@ import java.util.regex.Pattern;
 
 public class Jdk implements Buildable, Iterable<File> {
 
-    private static final List<String> ALLOWED_ARCHITECTURES = Collections.unmodifiableList(Arrays.asList("aarch64", "x64"));
+    private static final List<String> ALLOWED_ARCHITECTURES = Collections.unmodifiableList(Arrays.asList("aarch64", "x64", "s390x"));
     private static final List<String> ALLOWED_VENDORS = Collections.unmodifiableList(Arrays.asList("adoptopenjdk", "openjdk"));
     private static final List<String> ALLOWED_PLATFORMS = Collections.unmodifiableList(Arrays.asList("darwin", "linux", "windows", "mac"));
     private static final Pattern VERSION_PATTERN = Pattern.compile("(\\d+)(\\.\\d+\\.\\d+)?\\+(\\d+(?:\\.\\d+)?)(@([a-f0-9]{32}))?");
diff --git a/buildSrc/src/test/java/org/elasticsearch/gradle/JdkDownloadPluginTests.java b/buildSrc/src/test/java/org/elasticsearch/gradle/JdkDownloadPluginTests.java
index b1065715442..4e08fae0e06 100644
--- a/buildSrc/src/test/java/org/elasticsearch/gradle/JdkDownloadPluginTests.java
+++ b/buildSrc/src/test/java/org/elasticsearch/gradle/JdkDownloadPluginTests.java
@@ -84,7 +84,7 @@ public class JdkDownloadPluginTests extends GradleUnitTestCase {
             "11.0.2+33",
             "linux",
             "unknown",
-            "unknown architecture [unknown] for jdk [testjdk], must be one of [aarch64, x64]"
+            "unknown architecture [unknown] for jdk [testjdk], must be one of [aarch64, x64, s390x]"
         );
     }
 
diff --git a/distribution/archives/build.gradle b/distribution/archives/build.gradle
index 49ab50b1718..3aacb04fb3f 100644
--- a/distribution/archives/build.gradle
+++ b/distribution/archives/build.gradle
@@ -132,6 +132,13 @@ distribution_archives {
     }
   }
 
+  linuxS390xTar {
+    archiveClassifier = 'linux-s390x'
+    content {
+      archiveFiles(modulesFiles(false, 'linux-s390x'), 'tar', 'linux', 's390x', false, true)
+    }
+  }
+
   linuxTar {
     archiveClassifier = 'linux-x86_64'
     content {
@@ -146,6 +153,13 @@ distribution_archives {
     }
   }
 
+  ossLinuxS390xTar {
+    archiveClassifier = 'linux-s390x'
+    content {
+      archiveFiles(modulesFiles(true, 'linux-s390x'), 'tar', 'linux', 's390x', true, true)
+    }
+  }
+
   ossLinuxTar {
     archiveClassifier = 'linux-x86_64'
     content {
diff --git a/distribution/build.gradle b/distribution/build.gradle
index 3e905c3e880..5ca2a4fe78f 100644
--- a/distribution/build.gradle
+++ b/distribution/build.gradle
@@ -293,7 +293,7 @@ configure(subprojects.findAll { ['archives', 'packages'].contains(it.name) }) {
   // Setup all required JDKs
   project.jdks {
     ['darwin', 'windows', 'linux'].each { platform ->
-      (platform == 'linux' ? ['x64', 'aarch64'] : ['x64']).each { architecture ->
+      (platform == 'linux' ? ['x64', 'aarch64', 's390x'] : ['x64']).each { architecture ->
         "bundled_${platform}_${architecture}" {
           it.platform = platform
           it.version = VersionProperties.getBundledJdk(platform)
@@ -372,7 +372,7 @@ configure(subprojects.findAll { ['archives', 'packages'].contains(it.name) }) {
         } else {
           buildModules = buildDefaultModulesTaskProvider
         }
-        List excludePlatforms = ['linux-x86_64', 'linux-aarch64', 'windows-x86_64', 'darwin-x86_64']
+        List excludePlatforms = ['linux-x86_64', 'linux-aarch64', 'linux-s390x', 'windows-x86_64', 'darwin-x86_64']
         if (platform != null) {
           excludePlatforms.remove(excludePlatforms.indexOf(platform))
         } else {
diff --git a/distribution/docker/build.gradle b/distribution/docker/build.gradle
index 7e434e9469c..304c161cf76 100644
--- a/distribution/docker/build.gradle
+++ b/distribution/docker/build.gradle
@@ -20,15 +20,19 @@ testFixtures.useFixture()
 configurations {
   aarch64DockerSource
   dockerSource
+  s390xDockerSource
   aarch64OssDockerSource
+  s390xOssDockerSource
   ossDockerSource
   transformLog4jJar
 }
 
 dependencies {
   aarch64DockerSource project(path: ":distribution:archives:linux-aarch64-tar", configuration:"default")
+  s390xDockerSource project(path: ":distribution:archives:linux-s390x-tar")
   dockerSource project(path: ":distribution:archives:linux-tar", configuration:"default")
   aarch64OssDockerSource project(path: ":distribution:archives:oss-linux-aarch64-tar", configuration:"default")
+  s390xOssDockerSource project(path: ":distribution:archives:oss-linux-s390x-tar")
   ossDockerSource project(path: ":distribution:archives:oss-linux-tar", configuration:"default")
   transformLog4jJar project(path: ":distribution:docker:transform-log4j-config", configuration: "default")
 }
@@ -107,7 +111,7 @@ class SquashNewlinesFilter extends FilterReader {
 
 private static String buildPath(Architecture architecture, boolean oss, DockerBase base) {
   return 'build/' +
-    (architecture == Architecture.AARCH64 ? 'aarch64-' : '') +
+    (architecture == Architecture.AARCH64 ? 'aarch64-' : architecture == Architecture.S390X ? 's390x-' : '') +
     (oss ? 'oss-' : '') +
     (base == DockerBase.UBI ? 'ubi-' : '') +
     (base == DockerBase.UBI ? 'ubi-' : (base == DockerBase.IRON_BANK ? 'ironbank-' : '')) +
@@ -116,7 +120,7 @@ private static String buildPath(Architecture architecture, boolean oss, DockerBa
 
 private static String taskName(String prefix, Architecture architecture, boolean oss, DockerBase base, String suffix) {
   return prefix +
-    (architecture == Architecture.AARCH64 ? 'Aarch64' : '') +
+    (architecture == Architecture.AARCH64 ? 'Aarch64' : architecture == Architecture.S390X ? 's390x-' : '') +
     (oss ? 'Oss' : '') +
     (base == DockerBase.UBI ? 'Ubi' : (base == DockerBase.IRON_BANK ? 'IronBank' : '')) +
     suffix
@@ -197,6 +201,12 @@ void addCopyDockerContextTask(Architecture architecture, boolean oss, DockerBase
       } else {
         from configurations.aarch64DockerSource
       }
+    } else if ("s390x".equals(architecture)) {
+      if (oss){
+        from configurations.s390xOssDockerSource
+      } else {
+        from configurations.s390xDockerSource
+      }
     } else {
       if (oss) {
         from configurations.ossDockerSource
@@ -232,7 +242,7 @@ tasks.register("copyNodeKeyMaterial", Sync) {
 elasticsearch_distributions {
   Architecture.values().each { eachArchitecture ->
     Flavor.values().each { distroFlavor ->
-      "docker_$distroFlavor${ eachArchitecture == Architecture.AARCH64 ? '_aarch64' : '' }" {
+      "docker_$distroFlavor${ eachArchitecture == Architecture.AARCH64 ? '_aarch64' :  eachArchitecture == Architecture.S390X ? '_s390x' : '' }" {
         architecture = eachArchitecture
         flavor = distroFlavor
         type = 'docker'
diff --git a/distribution/packages/build.gradle b/distribution/packages/build.gradle
index f34a363cde1..61b2def6a6f 100644
--- a/distribution/packages/build.gradle
+++ b/distribution/packages/build.gradle
@@ -99,6 +99,8 @@ Closure commonPackageConfig(String type, boolean oss, boolean jdk, String archit
     if (type == 'deb') {
       if (architecture == 'x64') {
         arch('amd64')
+      } else if (architecture == 's390x') {
+        arch('s390x')
       } else {
         assert architecture == 'aarch64' : architecture
         arch('arm64')
@@ -107,6 +109,8 @@ Closure commonPackageConfig(String type, boolean oss, boolean jdk, String archit
       assert type == 'rpm' : type
       if (architecture == 'x64') {
         arch('X86_64')
+      } else if (architecture == 's390x') {
+        arch('s390x')
       } else {
         assert architecture == 'aarch64' : architecture
         arch('aarch64')
@@ -145,7 +149,7 @@ Closure commonPackageConfig(String type, boolean oss, boolean jdk, String archit
         with libFiles(oss)
       }
       into('modules') {
-        with modulesFiles(oss, 'linux-' + ((architecture == 'x64') ? 'x86_64' : architecture))
+        with modulesFiles(oss, 'linux-' + ((architecture == 'x64') ? 'x86_64' : (architecture == 's390x')? 's390x' : architecture))
       }
       if (jdk) {
         into('jdk') {
@@ -346,6 +350,17 @@ tasks.register('buildAarch64Deb', Deb) {
   configure(commonDebConfig(false, true, 'aarch64'))
 }
 
+
+tasks.register('buildS390xDeb', Deb) {
+  configure(commonDebConfig(false, true, 's390x'))
+}
+
+
+tasks.register('buildS390xOssDeb', Deb) {
+  configure(commonDebConfig(true, true, 's390x'))
+}
+
+
 tasks.register('buildDeb', Deb) {
   configure(commonDebConfig(false, true, 'x64'))
 }
@@ -399,6 +414,16 @@ tasks.register('buildAarch64Rpm', Rpm) {
   configure(commonRpmConfig(false, true, 'aarch64'))
 }
 
+
+tasks.register('buildS390xRpm', Rpm) {
+  configure(commonRpmConfig(false, true, 's390x'))
+}
+
+tasks.register('buildS390xOssRpm', Rpm) {
+  configure(commonRpmConfig(true, true, 's390x'))
+}
+
+
 tasks.register('buildRpm', Rpm) {
   configure(commonRpmConfig(false, true, 'x64'))
 }
diff --git a/server/build.gradle b/server/build.gradle
index 69343342d65..039080aab6b 100644
--- a/server/build.gradle
+++ b/server/build.gradle
@@ -108,8 +108,7 @@ dependencies {
   api "org.apache.logging.log4j:log4j-api:${versions.log4j}"
   api "org.apache.logging.log4j:log4j-core:${versions.log4j}", optional
 
-  // repackaged jna with native bits linked against all elastic supported platforms
-  api "org.elasticsearch:jna:${versions.jna}"
+  api "net.java.dev.jna:jna:${versions.jna}"
 
   testImplementation(project(":test:framework")) {
     // tests use the locally compiled version of server
diff --git a/server/licenses/jna-5.5.0.jar.sha1 b/server/licenses/jna-5.5.0.jar.sha1
index 903d2bd4cd9..8e673d6e70c 100644
--- a/server/licenses/jna-5.5.0.jar.sha1
+++ b/server/licenses/jna-5.5.0.jar.sha1
@@ -1 +1 @@
-ade077cbb2618a18bfc6c335413b2b7163d97601
\ No newline at end of file
+0e0845217c4907822403912ad6828d8e0b256208
\ No newline at end of file
diff --git a/settings.gradle b/settings.gradle
index eba6a72456e..b2af0022f38 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -27,14 +27,20 @@ List projects = [
   'distribution:archives:oss-no-jdk-darwin-tar',
   'distribution:archives:no-jdk-darwin-tar',
   'distribution:archives:oss-linux-aarch64-tar',
+  'distribution:archives:oss-linux-s390x-tar',
   'distribution:archives:oss-linux-tar',
   'distribution:archives:linux-aarch64-tar',
+  'distribution:archives:linux-s390x-tar',
   'distribution:archives:linux-tar',
   'distribution:archives:oss-no-jdk-linux-tar',
   'distribution:archives:no-jdk-linux-tar',
   'distribution:docker',
   'distribution:docker:docker-aarch64-build-context',
   'distribution:docker:docker-aarch64-export',
+  'distribution:docker:docker-s390x-build-context',
+  'distribution:docker:docker-s390x-export',
+  'distribution:docker:oss-docker-s390x-build-context',
+  'distribution:docker:oss-docker-s390x-export',
   'distribution:docker:docker-build-context',
   'distribution:docker:docker-export',
   'distribution:docker:ironbank-docker-build-context',
@@ -46,16 +52,21 @@ List projects = [
   'distribution:docker:ubi-docker-aarch64-export',
   'distribution:docker:ubi-docker-build-context',
   'distribution:docker:ubi-docker-export',
+  'distribution:docker:ubi-docker-s390x-export',
   'distribution:packages:aarch64-oss-deb',
   'distribution:packages:oss-deb',
   'distribution:packages:aarch64-deb',
+  'distribution:packages:s390x-oss-deb',
   'distribution:packages:deb',
   'distribution:packages:oss-no-jdk-deb',
+  'distribution:packages:s390x-deb',
   'distribution:packages:no-jdk-deb',
   'distribution:packages:aarch64-oss-rpm',
   'distribution:packages:oss-rpm',
+  'distribution:packages:s390x-oss-rpm',
   'distribution:packages:aarch64-rpm',
   'distribution:packages:rpm',
+  'distribution:packages:s390x-rpm',
   'distribution:packages:oss-no-jdk-rpm',
   'distribution:packages:no-jdk-rpm',
   'distribution:bwc:bugfix',
diff --git a/x-pack/plugin/sql/qa/server/build.gradle b/x-pack/plugin/sql/qa/server/build.gradle
index 82d8d5e5423..cc11caeb66a 100644
--- a/x-pack/plugin/sql/qa/server/build.gradle
+++ b/x-pack/plugin/sql/qa/server/build.gradle
@@ -32,7 +32,7 @@ dependencies {
   api "org.jline:jline-reader:${jlineVersion}"
   api "org.jline:jline-style:${jlineVersion}"
 
-  testRuntimeOnly "org.elasticsearch:jna:${versions.jna}"
+  testRuntimeOnly "net.java.dev.jna:jna:${versions.jna}"
 }
 
 // this is just a test fixture used by other projects and not in production
@@ -95,7 +95,7 @@ subprojects {
     testRuntimeOnly "org.jline:jline-reader:${jlineVersion}"
     testRuntimeOnly "org.jline:jline-style:${jlineVersion}"
 
-    testRuntimeOnly "org.elasticsearch:jna:${versions.jna}"
+    testRuntimeOnly "net.java.dev.jna:jna:${versions.jna}"
 
     // spatial dependency
     testRuntimeOnly project(path: xpackModule('spatial'))
diff --git a/x-pack/plugin/sql/sql-cli/build.gradle b/x-pack/plugin/sql/sql-cli/build.gradle
index a30a38b2e6d..73670c2c594 100644
--- a/x-pack/plugin/sql/sql-cli/build.gradle
+++ b/x-pack/plugin/sql/sql-cli/build.gradle
@@ -30,7 +30,7 @@ dependencies {
   api xpackProject('plugin:sql:sql-action')
   api project(":libs:elasticsearch-cli")
   api project(':libs:elasticsearch-x-content')
-  runtimeOnly "org.elasticsearch:jna:${versions.jna}"
+  runtimeOnly "net.java.dev.jna:jna:${versions.jna}"
   testImplementation project(":test:framework")
 }
 
diff --git a/x-pack/plugin/sql/sql-cli/licenses/jna-5.5.0.jar.sha1 b/x-pack/plugin/sql/sql-cli/licenses/jna-5.5.0.jar.sha1
index 903d2bd4cd9..8e673d6e70c 100644
--- a/x-pack/plugin/sql/sql-cli/licenses/jna-5.5.0.jar.sha1
+++ b/x-pack/plugin/sql/sql-cli/licenses/jna-5.5.0.jar.sha1
@@ -1 +1 @@
-ade077cbb2618a18bfc6c335413b2b7163d97601
\ No newline at end of file
+0e0845217c4907822403912ad6828d8e0b256208
\ No newline at end of file
