diff --git a/images/build/cross/Makefile b/images/build/cross/Makefile
index 0c1a2894..659e134f 100644
--- a/images/build/cross/Makefile
+++ b/images/build/cross/Makefile
@@ -28,9 +28,9 @@ IMGNAME = kube-cross
 # - v1.100.0-go1.17-bullseye.0 satisfies SemVer regex, while:
 # - v1.100-go1.17-bullseye.0 does not
 KUBERNETES_VERSION ?= v1.36.0
-GO_VERSION ?= 1.25.6
-GO_MAJOR_VERSION ?= 1.25
-OS_CODENAME ?= bullseye
+GO_VERSION ?= 1.26.2
+GO_MAJOR_VERSION ?= 1.26
+OS_CODENAME ?= trixie
 REVISION ?= 0
 TYPE ?= default

@@ -53,7 +53,7 @@ export DOCKER_CLI_EXPERIMENTAL=enabled
 # TODO: Support multi-arch kube-cross images for linux/arm
 #       Currently some of the components references in the Dockerfile are
 #       not supported in specific architectures
-PLATFORMS ?= linux/amd64 linux/arm64 #linux/arm
+PLATFORMS ?= linux/s390x

 KUBE_DYNAMIC_CROSSPLATFORMS ?= "arm64 armhf i386 ppc64el s390x"
 KUBE_CROSSPLATFORMS ?= "linux/386 linux/arm linux/arm64 linux/ppc64le linux/s390x darwin/amd64 windows/amd64 windows/386"
diff --git a/images/build/distroless-iptables/Makefile b/images/build/distroless-iptables/Makefile
index b0684a1d..81d45476 100644
--- a/images/build/distroless-iptables/Makefile
+++ b/images/build/distroless-iptables/Makefile
@@ -19,9 +19,9 @@ IMAGE=$(REGISTRY)/distroless-iptables

 TAG ?= $(shell git describe --tags --always --dirty)
 IMAGE_VERSION ?= v0.8.7
-CONFIG ?= distroless-bookworm
-BASEIMAGE ?= debian:bookworm-slim
-GORUNNER_VERSION ?= v2.4.0-go1.25.6-bookworm.0
+CONFIG ?= distroless-trixie
+BASEIMAGE ?= debian:trixie-slim
+GORUNNER_VERSION ?= v2.4.0-go1.25.6-trixie.0

 ARCH?=amd64
 ALL_ARCH = amd64 arm arm64 ppc64le s390x