diff --git a/cmake/modules/plugins.cmake b/cmake/modules/plugins.cmake
index 8a4a1d28..60c1280b 100644
--- a/cmake/modules/plugins.cmake
+++ b/cmake/modules/plugins.cmake
@@ -28,13 +28,13 @@ endif()
 
 ExternalProject_Add(
   k8saudit-plugin
-  URL "https://download.falco.org/plugins/stable/k8saudit-${PLUGIN_K8S_AUDIT_VERSION}-${PLUGINS_SYSTEM_NAME}-${CMAKE_HOST_SYSTEM_PROCESSOR}.tar.gz"
-  URL_HASH "SHA256=${PLUGIN_K8S_AUDIT_HASH}"
+  URL "https://github.com/falcosecurity/plugins/archive/refs/tags/k8saudit-0.3.0.tar.gz"
+  URL_HASH "SHA256=09a594f0a6471e39151b1908fccb855a52c6296a2593bd9f064313205e09b495"
+  SOURCE_DIR "${PROJECT_BINARY_DIR}/k8saudit-plugin-src"
+  BINARY_DIR "${PROJECT_BINARY_DIR}/k8saudit-plugin-build"
   CONFIGURE_COMMAND ""
-  BUILD_COMMAND ""
-  INSTALL_COMMAND "")
-
-install(FILES "${PROJECT_BINARY_DIR}/k8saudit-plugin-prefix/src/k8saudit-plugin/libk8saudit.so" DESTINATION "${FALCO_PLUGINS_DIR}" COMPONENT "${PLUGINS_COMPONENT_NAME}")
+  BUILD_COMMAND cd ${PROJECT_BINARY_DIR}/k8saudit-plugin-src/plugins/k8saudit && make
+  INSTALL_COMMAND ${CMAKE_COMMAND} -E copy ${PROJECT_BINARY_DIR}/k8saudit-plugin-src/plugins/k8saudit/libk8saudit.so ${FALCO_PLUGINS_DIR})
 
 ExternalProject_Add(
   k8saudit-rules
@@ -55,13 +55,15 @@ endif()
 
 ExternalProject_Add(
   cloudtrail-plugin
-  URL "https://download.falco.org/plugins/stable/cloudtrail-${PLUGIN_CLOUDTRAIL_VERSION}-${PLUGINS_SYSTEM_NAME}-${CMAKE_HOST_SYSTEM_PROCESSOR}.tar.gz"
-  URL_HASH "SHA256=${PLUGIN_CLOUDTRAIL_HASH}"
+  URL "https://github.com/falcosecurity/plugins/archive/refs/tags/cloudtrail-0.5.0.tar.gz"
+  URL_HASH "SHA256=6a3af4c3da0ab1ea9029247319866dc2054f2c0fb23e06116a4e96584122e750"
+  SOURCE_DIR "${PROJECT_BINARY_DIR}/cloudtrail-plugin-src"
+  BINARY_DIR "${PROJECT_BINARY_DIR}/cloudtrail-plugin-build"
   CONFIGURE_COMMAND ""
-  BUILD_COMMAND ""
-  INSTALL_COMMAND "")
+  BUILD_COMMAND cd ${PROJECT_BINARY_DIR}/cloudtrail-plugin-src/plugins/cloudtrail && make
+  INSTALL_COMMAND ${CMAKE_COMMAND} -E copy ${PROJECT_BINARY_DIR}/cloudtrail-plugin-src/plugins/cloudtrail/libcloudtrail.so ${FALCO_PLUGINS_DIR})
+
 
-install(FILES "${PROJECT_BINARY_DIR}/cloudtrail-plugin-prefix/src/cloudtrail-plugin/libcloudtrail.so" DESTINATION "${FALCO_PLUGINS_DIR}" COMPONENT "${PLUGINS_COMPONENT_NAME}")
 
 ExternalProject_Add(
   cloudtrail-rules
@@ -82,10 +84,12 @@ endif()
 
 ExternalProject_Add(
   json-plugin
-  URL "https://download.falco.org/plugins/stable/json-${PLUGIN_JSON_VERSION}-${PLUGINS_SYSTEM_NAME}-${CMAKE_HOST_SYSTEM_PROCESSOR}.tar.gz"
-  URL_HASH "SHA256=${PLUGIN_JSON_HASH}"
+  URL "https://github.com/falcosecurity/plugins/archive/refs/tags/json-0.5.0.tar.gz"
+  URL_HASH "SHA256=14f216cfcfec4cf50ab4ac208c5008ac206178253b9e2470cc60589569956bd0"
+  SOURCE_DIR "${PROJECT_BINARY_DIR}/json-plugin-src"
+  BINARY_DIR "${PROJECT_BINARY_DIR}/json-plugin-build"
   CONFIGURE_COMMAND ""
-  BUILD_COMMAND ""
-  INSTALL_COMMAND "")
+  BUILD_COMMAND cd ${PROJECT_BINARY_DIR}/json-plugin-src/plugins/json && make
+  INSTALL_COMMAND ${CMAKE_COMMAND} -E copy ${PROJECT_BINARY_DIR}/json-plugin-src/plugins/json/libjson.so ${FALCO_PLUGINS_DIR})
+
 
-install(FILES "${PROJECT_BINARY_DIR}/json-plugin-prefix/src/json-plugin/libjson.so" DESTINATION "${FALCO_PLUGINS_DIR}" COMPONENT "${PLUGINS_COMPONENT_NAME}")
