diff --git a/images/build/cross/Makefile b/images/build/cross/Makefile
index 554693ab..33dcd5ae 100644
--- a/images/build/cross/Makefile
+++ b/images/build/cross/Makefile
@@ -28,7 +28,7 @@ IMGNAME = kube-cross
 # - v1.100.0-go1.17-bullseye.0 satisfies SemVer regex, while:
 # - v1.100-go1.17-bullseye.0 does not
 KUBERNETES_VERSION ?= v1.31.0
-GO_VERSION ?= 1.23.2
+GO_VERSION ?= 1.23.4
 GO_MAJOR_VERSION ?= 1.23
-OS_CODENAME ?= bullseye
+OS_CODENAME ?= bookworm
 REVISION ?= 0
@@ -53,7 +53,7 @@ export DOCKER_CLI_EXPERIMENTAL=enabled
 # TODO: Support multi-arch kube-cross images for linux/arm
 #       Currently some of the components references in the Dockerfile are
 #       not supported in specific architectures
-PLATFORMS ?= linux/amd64 linux/arm64 #linux/arm
+PLATFORMS ?= linux/s390x

 # for legacy images only build linux/amd64
 ifeq ($(TYPE), legacy)
