--- a/Makefile
+++ b/Makefile
@@ -39,7 +39,7 @@ endif

 UNAME := $(shell uname)
 ifeq ($(UNAME),Linux)
-BAZEL_CONFIG_DEV  = --config=libc++
+BAZEL_CONFIG_DEV  =
 BAZEL_CONFIG_REL  = --config=libc++ --config=release
 BAZEL_CONFIG_ASAN = --config=clang-asan --config=libc++
 BAZEL_CONFIG_TSAN = --config=clang-tsan --config=libc++
@@ -61,7 +61,7 @@ clean:
        @bazel clean

 test:
-	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && bazel $(BAZEL_STARTUP_ARGS) test $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_DEV) $(BAZEL_TARGETS)
+	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && bazel $(BAZEL_STARTUP_ARGS) test --copt "-DENVOY_IGNORE_GLIBCXX_USE_CXX11_ABI_ERROR=1" $(BAZEL_TEST_ARGS) $(BAZEL_TARGETS) --test_env=ENVOY_IP_TEST_VERSIONS=v4only --host_javabase=@local_jdk//:jdk --test_env=HEAPCHECK=

 test_asan:
        export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && bazel $(BAZEL_STARTUP_ARGS) test $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_ASAN) -- $(BAZEL_TARGETS) $(SANITIZER_EXCLUSIONS)
