Index: DriverManager/SQLAllocHandle.c
<+>UTF-8
===================================================================
diff --git a/DriverManager/SQLAllocHandle.c b/DriverManager/SQLAllocHandle.c
--- a/DriverManager/SQLAllocHandle.c	(revision d43ff72004e0c194e4f9dad6e0c49cd356616702)
+++ b/DriverManager/SQLAllocHandle.c	(date 1670877160550)
@@ -483,26 +483,6 @@
 
             *output_handle = (SQLHANDLE) connection;
 
-#ifndef ENABLE_DRIVER_ICONV
-
-            /*
-             * initialize unicode
-             */
-
-            if ( !unicode_setup( connection ))
-            {
-                char txt[ 256 ];
-
-                sprintf( txt, "Can't initiate unicode conversion" );
-
-                dm_log_write( __FILE__,
-                        __LINE__,
-                        LOG_INFO,
-                        LOG_INFO,
-                        txt );
-            }
-#endif
-
             if ( log_info.log_flag )
             {
                 sprintf( environment -> msg, 
