diff --git a/libraries/cmake/source/aws-sdk-cpp/src/s2n/pq-crypto/sike_r3/sikep434r3.h b/libraries/cmake/source/aws-sdk-cpp/src/s2n/pq-crypto/sike_r3/sikep434r3.h
index 5b797b1..b0eb76c 100644
--- a/libraries/cmake/source/aws-sdk-cpp/src/s2n/pq-crypto/sike_r3/sikep434r3.h
+++ b/libraries/cmake/source/aws-sdk-cpp/src/s2n/pq-crypto/sike_r3/sikep434r3.h
@@ -33,7 +33,7 @@ uint64_t bswap64(uint64_t x);
 /* Arch specific definitions */
 #define digit_t S2N_SIKE_P434_R3_NAMESPACE(digit_t)
 #define hdigit_t S2N_SIKE_P434_R3_NAMESPACE(hdigit_t)
-#if defined(_AMD64_) || defined(__x86_64) || defined(__x86_64__) || defined(__aarch64__) || defined(_S390X_) || defined(_ARM64_) || defined(__powerpc64__) || (defined(__riscv) && (__riscv_xlen == 64))
+#if defined(_AMD64_) || defined(__x86_64) || defined(__x86_64__) || defined(__aarch64__) || defined(_S390X_) || defined(__s390x__) || defined(_ARM64_) || defined(__powerpc64__) || (defined(__riscv) && (__riscv_xlen == 64))
     #define S2N_SIKE_P434_R3_NWORDS_FIELD    7 /* Number of words of a 434-bit field element */
     #define S2N_SIKE_P434_R3_ZERO_WORDS      3 /* Number of "0" digits in the least significant part of p434 + 1 */
     #define S2N_SIKE_P434_R3_RADIX           64
