diff --git a/buf/internal/toolchain.bzl b/buf/internal/toolchain.bzl
index 1cc60ff..b612d29 100644
--- a/buf/internal/toolchain.bzl
+++ b/buf/internal/toolchain.bzl
@@ -115,13 +115,16 @@ def _buf_download_releases_impl(ctx):
         version = versions[0]["name"]
 
     os, cpu = _detect_host_platform(ctx)
-    if os not in ["linux", "darwin", "windows"] or cpu not in ["arm64", "amd64"]:
+    if version != "local" and (os not in ["linux", "darwin", "windows"] or cpu not in ["arm64", "amd64"]):
         fail("Unsupported operating system or cpu architecture ")
     if os == "linux" and cpu == "arm64":
         cpu = "aarch64"
     if cpu == "amd64":
         cpu = "x86_64"
 
+    if version == "local":
+        repository_url = ctx.os.environ.get("BUFBUILD_BUF_TOOLCHAIN_URL")
+
     ctx.report_progress("Downloading buf release hash")
     sha256 = ctx.download(
         url = [
