diff --git a/Makefile b/Makefile
index 6e19467e..2786044f 100644
--- a/Makefile
+++ b/Makefile
@@ -106,7 +106,7 @@ endif

 PACKAGE_NAME?=github.com/tigera/operator
 LOCAL_USER_ID?=$(shell id -u $$USER)
-GO_BUILD_VER?=v0.88
+GO_BUILD_VER?=v0.89
 CALICO_BUILD?=calico/go-build:$(GO_BUILD_VER)-$(ARCH)
 SRC_FILES=$(shell find ./pkg -name '*.go')
 SRC_FILES+=$(shell find ./api -name '*.go')
diff --git a/build/Dockerfile.s390x b/build/Dockerfile.s390x
index 6eaccc95..055de632 100644
--- a/build/Dockerfile.s390x
+++ b/build/Dockerfile.s390x
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.

-ARG QEMU_IMAGE=calico/go-build:v0.84-s390x
+ARG QEMU_IMAGE=calico/go-build:v0.89-s390x
 FROM ${QEMU_IMAGE} as qemu

 FROM registry.access.redhat.com/ubi8/ubi-minimal:8.8 AS ubi
@@ -34,6 +34,8 @@ COPY --from=qemu /usr/bin/qemu-*-static /usr/bin/
 COPY --from=ubi /licenses /licenses
 COPY --from=ubi /etc/pki /etc/pki
 COPY --from=ubi /usr/share/pki /usr/share/pki
+COPY --from=ubi /lib /lib
+COPY --from=ubi /lib64 /lib64

 ARG GIT_VERSION=unknown
 LABEL name="Tigera Operator" \

