diff --git a/deploy/Dockerfile b/deploy/Dockerfile
index 37c4a059..01ad4e82 100644
--- a/deploy/Dockerfile
+++ b/deploy/Dockerfile
@@ -32,7 +32,7 @@ RUN GOARCH=$(cat /goarch) && CGO_ENABLED=0 && \
   cd /go/src/github.com/GoogleCloudPlatform                                  && \
   git clone https://github.com/GoogleCloudPlatform/docker-credential-gcr.git && \
   cd /go/src/github.com/GoogleCloudPlatform/docker-credential-gcr            && \
-  make deps OUT_DIR=/usr/local/bin                                           && \
+  go get -u -t ./...                                                         && \
   go build -ldflags "-linkmode external -extldflags -static" -i -o /usr/local/bin/docker-credential-gcr main.go
 
 # Get Amazon ECR credential helper
@@ -50,7 +50,7 @@ RUN GOARCH=$(cat /goarch) && (mkdir -p /go/src/github.com/chrismellard || true)
 RUN mkdir -p /kaniko/.docker
 
 COPY . .
-RUN make GOARCH=$(cat /goarch.txt)
+RUN make GOARCH=$(cat /goarch)
 
 # Generate latest ca-certificates
 
diff --git a/hack/install_golint.sh b/hack/install_golint.sh
index d0b06977..64e00fb4 100755
--- a/hack/install_golint.sh
+++ b/hack/install_golint.sh
@@ -70,6 +70,7 @@ is_supported_platform() {
     linux/amd64) found=0 ;;
     linux/386) found=0 ;;
     linux/ppc64le) found=0 ;;
+    linux/s390x) found=0 ;;
   esac
   return $found
 }
