--- /home/test/BUILD-exe_old	2019-05-29 11:15:56.157792527 +0000
+++ source/exe/BUILD	2019-05-29 11:17:09.557772508 +0000
@@ -45,6 +45,7 @@
     ] + select({
         "//bazel:windows_x86_64": envoy_windows_extensions(),
         "//bazel:linux_ppc": envoy_all_extensions(PPC_SKIP_TARGETS),
+        "//bazel:linux_s390x": envoy_all_extensions(),
         "//conditions:default": envoy_all_extensions(),
     }),
 )
