--- ./tools/cpp/lib_cc_configure.bzl.oldfile    2019-05-09 06:09:31.045362405 -0400
+++ ./tools/cpp/lib_cc_configure.bzl.newfile    2019-05-09 06:09:09.902469492 -0400
@@ -188,6 +188,8 @@
     result = repository_ctx.execute(["uname", "-m"])
     if result.stdout.strip() in ["power", "ppc64le", "ppc", "ppc64"]:
         return "ppc"
+    if result.stdout.strip() in ["s390x"]:
+        return "s390x"
     if result.stdout.strip() in ["arm", "armv7l"]:
         return "arm"
     if result.stdout.strip() in ["aarch64"]:
