--- Makefile.oldfile	2019-05-15 04:42:13.834528065 +0000
+++ Makefile.newfile	2019-05-15 04:46:08.864528065 +0000
@@ -34,7 +34,7 @@
 PATH := /usr/lib/llvm-7/bin:$(PATH)
 
 build:
-	PATH=$(PATH) CC=$(CC) CXX=$(CXX) bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) $(BAZEL_TARGETS) --host_javabase=@local_jdk//:jdk
+	PATH=$(PATH) CC=$(CC) CXX=$(CXX) bazel $(BAZEL_STARTUP_ARGS) build -c opt $(BAZEL_BUILD_ARGS) $(BAZEL_TARGETS) --host_javabase=@local_jdk//:jdk
 	@bazel shutdown
 
 # Build only envoy - fast
