--- Makefile.oldfile	2019-05-15 04:42:13.834528065 +0000
+++ Makefile.newfile	2019-05-15 04:52:56.414528065 +0000
@@ -47,7 +47,7 @@
 	@bazel shutdown
 
 test:
-	PATH=$(PATH) CC=$(CC) CXX=$(CXX) bazel $(BAZEL_STARTUP_ARGS) test $(BAZEL_TEST_ARGS) $(BAZEL_TARGETS)
+	PATH=$(PATH) CC=$(CC) CXX=$(CXX) bazel $(BAZEL_STARTUP_ARGS) test $(BAZEL_TEST_ARGS) $(BAZEL_TARGETS) --test_env=ENVOY_IP_TEST_VERSIONS=v4only --host_javabase=@local_jdk//:jdk
 	@bazel shutdown
 
 test_asan:
