--- WORKSPACE.oldfile   2019-05-22 07:21:52.372213557 +0000
+++ WORKSPACE.newfile   2019-05-29 11:32:53.995667302 +0000
@@ -31,22 +31,32 @@
     name = "boringssl_crypto",
     actual = "//external:ssl",
 )
-
+local_repository(
+     name = "boringssl",
+     path = "$SOURCE_ROOT/boringssl",
+)
+bind(
+     name = "ssl",
+     actual = "@boringssl//:ssl",
+)
 # envoy commit date 04/10/2019
 # bazel version: 0.22.0

 # When updating envoy sha manually please update the sha in istio.deps file also
 #
 # Determine SHA256 `wget https://github.com/envoyproxy/envoy/archive/COMMIT.tar.gz && sha256sum COMMIT.tar.gz`
-ENVOY_SHA = "ac7aa5ac8a815e5277b4d4659c5c02145fa1d56f"
+#ENVOY_SHA = "ac7aa5ac8a815e5277b4d4659c5c02145fa1d56f"
 ENVOY_SHA256 = "3f13facc893ef0c5063c7391a1ffca8de0f52425c8a7a49ef45e69dbb5e7304b"
 LOCAL_ENVOY_PROJECT = "/PATH/TO/ENVOY"

-http_archive(
+local_repository(
     name = "envoy",
-    strip_prefix = "envoy-" + ENVOY_SHA,
-    url = "https://github.com/envoyproxy/envoy/archive/" + ENVOY_SHA + ".tar.gz",
-    sha256 = ENVOY_SHA256,
+    path = "$SOURCE_ROOT/envoy"
+)
+
+local_repository(
+    name = "rules_foreign_cc",
+    path = "$SOURCE_ROOT/rules_foreign_cc",
 )

 # TODO(silentdai) Use bazel args to select envoy between local or http
