diff --git a/crypto/blake2/blake2.cc b/crypto/blake2/blake2.cc
--- a/crypto/blake2/blake2.cc
+++ b/crypto/blake2/blake2.cc
@@ -161,5 +161,7 @@ void BLAKE2B256_Final(uint8_t out[BLAKE2B256_DIGEST_LENGTH], BLAKE2B_CTX *b2b) {
   blake2b_transform(b2b, b2b->block, b2b->block_used,
                     /*is_final_block=*/1);
   static_assert(BLAKE2B256_DIGEST_LENGTH <= sizeof(b2b->h));
-  memcpy(out, b2b->h, BLAKE2B256_DIGEST_LENGTH);
+  for (int i = 0; i < BLAKE2B256_DIGEST_LENGTH / 8; ++i) {
+    CRYPTO_store_u64_le(out + 8 * i, b2b->h[i]);
+  }
 }
diff --git a/include/openssl/target.h b/include/openssl/target.h
--- a/include/openssl/target.h
+++ b/include/openssl/target.h
@@ -57,3 +57,6 @@
 #elif defined(__myriad2__)
 #define OPENSSL_32_BIT
+#elif defined(__s390x__)
+#define OPENSSL_64_BIT
+#define OPENSSL_S390X
 #else
