diff --git a/BUILD.bazel b/BUILD.bazel
index 8994f79f2..3e3bd34d8 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -386,6 +386,14 @@ config_setting(
     ],
 )
 
+config_setting(
+    name = "linux-s390_64",
+    constraint_values = [
+	"@platforms//os:linux",
+	"@platforms//cpu:s390x",
+    ],
+)
+
 config_setting(
     name = "osx-aarch_64",
     constraint_values = [
@@ -417,6 +425,7 @@ alias(
         ":linux-aarch_64": "@com_google_protobuf_protoc_linux_aarch_64//:protoc",
         ":linux-x86_64": "@com_google_protobuf_protoc_linux_x86_64//:protoc",
         ":linux-ppcle_64": "@com_google_protobuf_protoc_linux_ppcle_64//:protoc",
+        ":linux-s390_64": "@com_google_protobuf_protoc_linux_s390_64//:protoc",
         ":osx-aarch_64": "@com_google_protobuf_protoc_osx_aarch_64//:protoc",
         ":osx-x86_64": "@com_google_protobuf_protoc_osx_x86_64//:protoc",
         ":win64": "@com_google_protobuf_protoc_win64//:protoc",
diff --git a/upb/wire/decode.c b/upb/wire/decode.c
index 6da57877b..b9ff2a075 100644
--- a/upb/wire/decode.c
+++ b/upb/wire/decode.c
@@ -1079,6 +1079,7 @@ const char* _upb_Decoder_DecodeWireValue(upb_Decoder* d, const char* ptr,
           *op = kUpb_DecodeOp_UnknownField;
           return ptr;
         }
+        _upb_Decoder_MungeInt32(val);
       } else {
         _upb_Decoder_Munge(field, val);
       }
