diff --git a/toolchain/cc_toolchain_config.bzl b/toolchain/cc_toolchain_config.bzl
--- a/toolchain/cc_toolchain_config.bzl
+++ b/toolchain/cc_toolchain_config.bzl
@@ -145,3 +145,12 @@
         ),
+        "linux-s390x": (
+            "clang-s390x-linux",
+            "s390x",
+            "glibc_unknown",
+            "clang",
+            "clang",
+            "glibc_unknown",
+            "s390x-linux-gnu",
+        ),
         "linux-x86_64": (
             "clang-x86_64-linux",
             "k8",
diff --git a/toolchain/internal/common.bzl b/toolchain/internal/common.bzl
--- a/toolchain/internal/common.bzl
+++ b/toolchain/internal/common.bzl
@@ -17,6 +17,7 @@ SUPPORTED_TARGETS = [
     ("linux", "x86_64"),
     ("linux", "aarch64"),
     ("linux", "armv7"),
+    ("linux", "s390x"),
     ("linux", "riscv64"),
     ("darwin", "x86_64"),
     ("darwin", "aarch64"),
diff --git a/toolchain/internal/configure.bzl b/toolchain/internal/configure.bzl
--- a/toolchain/internal/configure.bzl
+++ b/toolchain/internal/configure.bzl
@@ -163,2 +163,3 @@
     if not toolchain_root:
-        fail("LLVM toolchain root missing for ({}, {})".format(os, arch))
+        _empty_repository(rctx)
+        return None
@@ -482,6 +483,7 @@
         "linux-aarch64": "aarch64-unknown-linux-gnu",
         "linux-armv7": "armv7-unknown-linux-gnueabihf",
         "linux-x86_64": "x86_64-unknown-linux-gnu",
+        "linux-s390x": "s390x-unknown-linux-gnu",
         "linux-riscv64": "riscv64-unknown-linux-gnu",
         "none-riscv32": "riscv32-unknown-none-elf",
         "none-riscv64": "riscv64-unknown-none-elf",
