diff --git a/CMakeLists.txt b/CMakeLists.txt
index 51f9e71..8fe83e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,7 +75,7 @@ endif()
 
 # Modern BPF is not supported on not Linux systems and in MINIMAL_BUILD
 if(CMAKE_SYSTEM_NAME MATCHES "Linux" AND NOT MINIMAL_BUILD)
-	option(BUILD_FALCO_MODERN_BPF "Build modern BPF support for Falco" ON)
+	option(BUILD_FALCO_MODERN_BPF "Build modern BPF support for Falco" OFF)
 	if(BUILD_FALCO_MODERN_BPF)
 		add_definitions(-DHAS_MODERN_BPF)
 	endif()
