--- Python-3.10.6/Lib/test/test_zlib.py.orig    2022-08-29 17:17:51.236654196 +0000
+++ Python-3.10.6/Lib/test/test_zlib.py 2022-08-29 17:19:28.526654196 +0000
@@ -3,6 +3,7 @@
 from test.support import import_helper
 import binascii
 import copy
+import os
 import pickle
 import random
 import sys
@@ -18,6 +19,8 @@
         hasattr(zlib.decompressobj(), "copy"),
         'requires Decompress.copy()')
 
+skip_on_s390x = unittest.skipIf(hasattr(os, 'uname') and os.uname().machine == 's390x',
+        'skipped on s390x')
 
 class VersionTestCase(unittest.TestCase):
 
@@ -182,6 +185,7 @@
                                          bufsize=zlib.DEF_BUF_SIZE),
                          HAMLET_SCENE)
 
+    @skip_on_s390x
     def test_speech128(self):
         # compress more data
         data = HAMLET_SCENE * 128
@@ -233,6 +237,7 @@
 
 class CompressObjectTestCase(BaseCompressTestCase, unittest.TestCase):
     # Test compression object
+    @skip_on_s390x
     def test_pair(self):
         # straightforward compress/decompress objects
         datasrc = HAMLET_SCENE * 128
