diff --git a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
index d02d9d417c9..f8824c2eb12 100644
--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
@@ -1062,7 +1062,16 @@ class BuildPlugin implements Plugin<Project> {
     }
 
     private static int numberOfPhysicalCores(Project project) {
-        if (project.file("/proc/cpuinfo").exists()) {
+        if ('s390x'.equals(System.getProperty('os.arch'))) {
+         // Ask s390x to count physical CPUs for us
+         ByteArrayOutputStream stdout = new ByteArrayOutputStream();
+         project.exec{
+             executable 'nproc'
+             args '--all'
+             standardOutput = stdout
+         }
+            return Integer.parseInt(stdout.toString('UTF-8').trim())
+        } else if (project.file("/proc/cpuinfo").exists()) {
             // Count physical cores on any Linux distro ( don't count hyper-threading )
             Map<String, Integer> socketToCore = [:]
             String currentID = ""
