diff --git a/Dockerfile b/Dockerfile
index 9186a05..6fb6f8e 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -4,12 +4,12 @@
 ARG JAVA_MAJOR
 ARG DISTRIB_NAME=rockylinux
 ARG DISTRIB_MAJOR
-ARG IMAGE_JAVA_REPO=quay.io/alfresco
+ARG IMAGE_JAVA_REPO=alfresco
 ARG IMAGE_JAVA_NAME=alfresco-base-java
 ARG IMAGE_JAVA_TAG=jre${JAVA_MAJOR}-${DISTRIB_NAME}${DISTRIB_MAJOR}
 
 # Tomcat is downloaded and configured on debian as its a binary dist anyway
-FROM debian:12-slim AS tomcat_dist
+FROM debian:trixie-slim AS tomcat_dist
 
 ARG TOMCAT_MAJOR
 ARG TOMCAT_VERSION
@@ -108,39 +108,31 @@ COPY --from=tomcat_dist /build/apr $BUILD_DIR/apr
 
 SHELL ["/bin/bash", "-euo", "pipefail", "-c"]
 
-RUN <<EOT
-  [ $DISTRIB_MAJOR -eq 8 ] && dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-${DISTRIB_MAJOR}.noarch.rpm
-  dnf install -y gcc make expat-devel java-${JAVA_MAJOR}-openjdk-devel redhat-rpm-config
+RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-${DISTRIB_MAJOR}.noarch.rpm && \
+  dnf install -y gcc make expat-devel java-${JAVA_MAJOR}-openjdk-devel redhat-rpm-config && \
   dnf clean all
-EOT
 
 WORKDIR ${BUILD_DIR}/apr
-RUN <<EOT
-  ./configure --prefix=${INSTALL_DIR}/apr
-  make -j "$(nproc)"
+RUN ./configure --prefix=${INSTALL_DIR}/apr && \
+  make -j "$(nproc)" && \
   make install
-EOT
 
 WORKDIR ${BUILD_DIR}/tcnative/native
-RUN <<EOT
-  if [ $DISTRIB_MAJOR -eq 8 ]; then
-    dnf install -y dnf-plugins-core
-    dnf config-manager -y --set-enabled powertools
-    dnf install -y openssl3-devel
-    ln -s /usr/include/openssl3/openssl /usr/include/openssl
-    export LIBS="-L/usr/lib64/openssl3 -Wl,-rpath,/usr/lib64/openssl3 -lssl -lcrypto"
-    export CFLAGS="-I/usr/include/openssl3"
-  else dnf install -y openssl-devel
-  fi
-  dnf clean all
-  ./configure \
-    --libdir=${INSTALL_DIR}/tcnative \
-    --with-apr=${INSTALL_DIR}/apr/bin/apr-1-config \
-    --with-java-home="$JAVA_HOME" \
-    --disable-openssl-version-check
-  make -j "$(nproc)"
+RUN if [ $DISTRIB_MAJOR -eq 8 ]; then \
+    dnf install -y dnf-plugins-core && \
+    dnf config-manager -y --set-enabled powertools && \
+    dnf install -y openssl3-devel && \
+    ln -s /usr/include/openssl3/openssl /usr/include/openssl && \
+    export LIBS="-L/usr/lib64/openssl3 -Wl,-rpath,/usr/lib64/openssl3 -lssl -lcrypto" && \
+    export CFLAGS="-I/usr/include/openssl3"; \
+  else \
+    dnf install -y openssl-devel; \
+  fi && \
+  dnf clean all	&& \
+  ./configure --libdir=${INSTALL_DIR}/tcnative --with-apr=${INSTALL_DIR}/apr/bin/apr-1-config --with-java-home="$JAVA_HOME" --disable-openssl-version-check && \
+  make -j "$(nproc)" && \
   make install
-EOT
+
 
 # hadolint ignore=DL3006
 FROM tcnative_build-${DISTRIB_NAME} AS tcnative_build
@@ -186,30 +178,30 @@ WORKDIR $CATALINA_HOME
 RUN groupadd --system tomcat && \
   useradd -M -s /bin/false --home $CATALINA_HOME --system --gid tomcat tomcat
 
-COPY --chown=:tomcat --chmod=640 --from=tomcat_dist /build/tomcat $CATALINA_HOME
-COPY --chown=:tomcat --chmod=640 --from=tcnative_build /usr/local/tcnative $TOMCAT_NATIVE_LIBDIR
-COPY --chown=:tomcat --chmod=640 --from=tcnative_build /usr/local/apr/lib/libapr-1.so* $APR_LIBDIR/
+COPY --from=tomcat_dist /build/tomcat $CATALINA_HOME
+COPY --from=tcnative_build /usr/local/tcnative $TOMCAT_NATIVE_LIBDIR
+COPY --from=tcnative_build /usr/local/apr/lib/libapr-1.so* $APR_LIBDIR/
+
 
 SHELL ["/bin/bash", "-euo", "pipefail", "-c"]
 
-RUN <<EOT
-  if [ $DISTRIB_MAJOR -eq 8 ]; then
-    dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
-    dnf install -y openssl3-libs
-    dnf clean all
-  fi
-  mkdir -m 770 logs temp work && chgrp tomcat . logs temp work
-  chmod ug+x bin/*.sh
-  find . -type d -exec chmod 770 {} +
-
-  echo "Testing Tomcat Native library..."
-  configtest_rc=0
-  configtest_out="$(catalina.sh configtest 2>&1)" || configtest_rc=$?
-  echo "$configtest_out"
-  [ $configtest_rc -eq 0 ] || { echo "catalina.sh configtest exited with code $configtest_rc"; exit $configtest_rc; }
+RUN if [ $DISTRIB_MAJOR -eq 8 ]; then \
+    dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && \
+    dnf install -y openssl3-libs && \
+    dnf clean all; \
+  fi && \
+  mkdir -m 770 logs temp work && chgrp tomcat . logs temp work && \
+  chmod ug+x bin/*.sh && \
+  find . -type d -exec chmod 770 {} + && \
+
+  echo "Testing Tomcat Native library..." && \
+  configtest_rc=0 && \
+  configtest_out="$(catalina.sh configtest 2>&1)" || configtest_rc=$? && \
+  echo "$configtest_out" && \
+  [ $configtest_rc -eq 0 ] || { echo "catalina.sh configtest exited with code $configtest_rc"; exit $configtest_rc; } && \
   echo "$configtest_out" | grep -q 'Loaded Apache Tomcat Native library' || \
     { echo "Tomcat Native library not found or not working properly"; exit 1; }
-EOT
+
 
 USER tomcat
 EXPOSE 8080
