diff --git a/docker-image/Dockerfile.s390x b/docker-image/Dockerfile.s390x
index f3b5373..6ed11ef 100644
--- a/docker-image/Dockerfile.s390x
+++ b/docker-image/Dockerfile.s390x
@@ -8,7 +8,7 @@ COPY --from=qemu /usr/bin/qemu-s390x-static /usr/bin/
 RUN apt-get update && \
 apt-get upgrade -y && \
 apt-get install -y --no-install-recommends \
-    gpg gpg-agent libelf-dev libmnl-dev libc-dev iptables libgcc-8-dev \
+    gpg gpg-agent libelf-dev libmnl-dev libc-dev iptables libgcc-8-dev zlib1g zlib1g-dev zlibc \
     bash-completion binutils binutils-dev ca-certificates make git curl \
     xz-utils gcc pkg-config bison flex build-essential && \
 apt-get purge --auto-remove && \
@@ -17,11 +17,14 @@ apt-get clean
 WORKDIR /tmp
 
 RUN \
-git clone --depth 1 -b master git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git && \
-cd linux/tools/bpf/bpftool/ && \
+git clone -b master git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git && \
+cd linux && \
+git checkout 222ee9087a730b1df08d09baed0d03626e67600f && \
+cd tools/bpf/bpftool/ && \
 sed -i '/CFLAGS += -O2/a CFLAGS += -static' Makefile && \
 sed -i 's/LIBS = -lelf $(LIBBPF)/LIBS = -lelf -lz $(LIBBPF)/g' Makefile && \
 printf 'feature-libbfd=0\nfeature-libelf=1\nfeature-bpf=1\nfeature-libelf-mmap=1' >> FEATURES_DUMP.bpftool && \
+printf '\nfeature-zlib=1' >> FEATURES_DUMP.bpftool && \
 FEATURES_DUMP=`pwd`/FEATURES_DUMP.bpftool make -j `getconf _NPROCESSORS_ONLN` && \
 strip bpftool && \
 ldd bpftool 2>&1 | grep -q -e "Not a valid dynamic program" \
