--- Gemfile.cp	2019-02-20 15:24:19.103437111 +0300
+++ Gemfile.orig	2019-02-20 15:23:24.513746450 +0300
@@ -7,12 +7,12 @@
 gem 'rugged', '~> 0.27'
 gem 'github-linguist', '~> 6.1', require: 'linguist'
 gem 'gitlab-markup', '~> 1.6.5'
-#gem 'gitaly-proto', '~> 1.5.0'
+gem 'gitaly-proto', '~> 1.5.0'
 gem 'activesupport', '~> 5.0.2'
 gem 'rdoc', '~> 4.2'
 gem 'gitlab-gollum-lib', '~> 4.2', require: false
 gem 'gitlab-gollum-rugged_adapter', '~> 0.4.4', require: false
-#gem 'grpc', '~> 1.15.0'
+gem 'grpc', '~> 1.15.0'
 gem 'sentry-raven', '~> 2.7.2', require: false
 gem 'faraday', '~> 0.12'
 
@@ -20,7 +20,7 @@
 # This version needs to be in sync with GitLab CE/EE
 gem 'licensee', '~> 8.9.0'
 
-#gem 'google-protobuf', '~> 3.6'
+gem 'google-protobuf', '~> 3.6'
 
 group :development, :test do
   gem 'rubocop', '~> 0.50', require: false
