--- Makefile_ORIG	2019-02-18 05:09:35.203419674 -0500
+++ Makefile	2019-02-19 03:51:53.302244930 -0500
@@ -11,7 +11,6 @@
 # Developer Tools
 PROTOC = $(TARGET_DIR)/protoc/bin/protoc
 PROTOC_GEN_GO = $(BIN_BUILD_DIR)/protoc-gen-go
-PROTOC_GEN_DOC = $(BIN_BUILD_DIR)/protoc-gen-doc
 
 .PHONY: all
 all: generate
@@ -43,13 +42,6 @@
 .PHONY: install-developer-tools
 install-developer-tools: $(TARGET_SETUP) $(PROTOC) .ruby-bundle
 
-.PHONY: docs
-docs: $(TARGET_SETUP) $(PROTOC_GEN_DOC) $(PROTOC)
-	rm -rf public && mkdir public
-	$(PROTOC) --doc_out=./public \
-	          --plugin=protoc-gen-doc=$(PROTOC_GEN_DOC) \
-	          *.proto
-
 $(PROTOC): $(TARGET_SETUP)
 	_support/install-protoc
 
@@ -57,5 +49,3 @@
 	bundle install --gemfile=_support/Gemfile --binstubs=$(BIN_BUILD_DIR)
 	touch $@	
 
-$(PROTOC_GEN_DOC): $(TARGET_SETUP)
-	go get -v -u github.com/pseudomuto/protoc-gen-doc/cmd/...
