--- Rakefile.orig	2019-02-04 04:04:04.614294000 -0500
+++ Rakefile	2019-02-04 04:18:43.421398000 -0500
@@ -34,7 +34,7 @@
     output_file = "lib/" + proto_file.sub(/\.proto$/, "_pb.rb")
     genproto_output << output_file
     file output_file => input_file do |file_task|
-      sh "../src/protoc -I../src --ruby_out=lib #{input_file}"
+      sh "./src/protoc -I../src --ruby_out=lib #{input_file}"
     end
   end
 end
@@ -64,7 +64,7 @@
     ext.cross_platform = [
       'x86-mingw32', 'x64-mingw32',
       'x86_64-linux', 'x86-linux',
-      'universal-darwin'
+      'universal-darwin', 's390x-linux'
     ]
   end
 
@@ -89,15 +89,15 @@
 genproto_output << "tests/test_import.rb"
 genproto_output << "tests/test_ruby_package.rb"
 file "tests/generated_code.rb" => "tests/generated_code.proto" do |file_task|
-  sh "../src/protoc --ruby_out=. tests/generated_code.proto"
+  sh "./src/protoc --ruby_out=. tests/generated_code.proto"
 end
 
 file "tests/test_import.rb" => "tests/test_import.proto" do |file_task|
-  sh "../src/protoc --ruby_out=. tests/test_import.proto"
+  sh "./src/protoc --ruby_out=. tests/test_import.proto"
 end
 
 file "tests/test_ruby_package.rb" => "tests/test_ruby_package.proto" do |file_task|
-  sh "../src/protoc --ruby_out=. tests/test_ruby_package.proto"
+  sh "./src/protoc --ruby_out=. tests/test_ruby_package.proto"
 end
 
 task :genproto => genproto_output
