--- upb.h.orig	2019-02-04 04:02:21.801869562 -0500
+++ upb.h	2019-02-04 03:59:27.215503868 -0500
@@ -13,6 +13,7 @@
 #include <stdarg.h>
 #include <stdbool.h>
 #include <stddef.h>
+#include <byteswap.h>
 
 #ifdef __cplusplus
 namespace upb {
@@ -8515,6 +8516,7 @@
   UPB_ASSERT(bytes <= 5);
   memcpy(&ret, buf, bytes);
+  ret = bswap_64(ret);
   UPB_ASSERT(ret <= 0xffffffffffU);
   return ret;
 }
 
