--- lib-es5/build_old.js        2019-01-28 08:45:14.447920625 +0000
+++ lib-es5/build.js    2019-01-28 08:45:43.197920625 +0000
@@ -195,7 +195,7 @@
           case 0:
           args = [];
           cpu = { x86: 'ia32', x64: 'x64',
-            armv6: 'arm', armv7: 'arm', arm64: 'arm64' }[targetArch];
+            armv6: 'arm', armv7: 'arm', arm64: 'arm64', s390x: 's390x' }[targetArch];

           args.push('--dest-cpu', cpu);
           // first of all v8_inspector introduces the use
