diff --git a/Makefile b/Makefile
index 0cbcced..2c33106 100644
--- a/Makefile
+++ b/Makefile
@@ -290,7 +290,7 @@ run-kubernetes-master: stop-kubernetes-master
 	docker run \
 		--net=host --name st-apiserver-${KUBE_APISERVER_PORT} \
 		--detach \
-		gcr.io/google_containers/hyperkube-amd64:${K8S_VERSION} kube-apiserver \
+		gcr.io/google_containers/hyperkube-${ARCH}:${K8S_VERSION} kube-apiserver \
 			--bind-address=0.0.0.0 \
 			--secure-port=1${KUBE_APISERVER_PORT} \
 			--insecure-bind-address=0.0.0.0 \
@@ -308,7 +308,7 @@ run-kubernetes-master: stop-kubernetes-master
 	docker run \
 		--net=host --name st-controller-manager-${KUBE_APISERVER_PORT} \
 		--detach \
-		gcr.io/google_containers/hyperkube-amd64:${K8S_VERSION} kube-controller-manager \
+		gcr.io/google_containers/hyperkube-${ARCH}:${K8S_VERSION} kube-controller-manager \
                         --master=127.0.0.1:${KUBE_APISERVER_PORT} \
                         --min-resync-period=3m \
                         --allocate-node-cidrs=true \
@@ -320,7 +320,7 @@ run-kubernetes-master: stop-kubernetes-master
 	    --net=host \
 	    --rm \
 		-v $(CURDIR):/manifests \
-		gcr.io/google_containers/hyperkube-amd64:${K8S_VERSION} kubectl \
+		gcr.io/google_containers/hyperkube-${ARCH}:${K8S_VERSION} kubectl \
 		--server=http://127.0.0.1:${KUBE_APISERVER_PORT} \
 		apply -f /manifests/tests/st/manifests/${KUBE_MOCK_NODE_MANIFEST}; \
 		do echo "Waiting for node to apply successfully..."; sleep 2; done
@@ -329,7 +329,7 @@ run-kubernetes-master: stop-kubernetes-master
 	-docker run \
 	    --net=host \
 	    --rm \
-		gcr.io/google_containers/hyperkube-amd64:${K8S_VERSION} kubectl \
+		gcr.io/google_containers/hyperkube-${ARCH}:${K8S_VERSION} kubectl \
 		--server=http://127.0.0.1:${KUBE_APISERVER_PORT} \
 		create namespace test
 	
diff --git a/tests/fv/utils/calicoctl.go b/tests/fv/utils/calicoctl.go
index bc78ea3..bb4f867 100644
--- a/tests/fv/utils/calicoctl.go
+++ b/tests/fv/utils/calicoctl.go
@@ -23,7 +23,7 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-var calicoctl = "/go/src/github.com/projectcalico/calicoctl/bin/calicoctl-linux-amd64"
+var calicoctl = "/go/src/github.com/projectcalico/calicoctl/bin/calicoctl-linux-s390x"
 
 func Calicoctl(kdd bool, args ...string) string {
 	out, err := CalicoctlMayFail(kdd, args...)
diff --git a/tests/st/utils/utils.py b/tests/st/utils/utils.py
index e247a11..6dc93fc 100644
--- a/tests/st/utils/utils.py
+++ b/tests/st/utils/utils.py
@@ -221,7 +221,7 @@ def calicoctl(command, data=None, load_as_stdin=False, format="yaml", only_stdou
     elif data and not load_as_stdin:
         option_file = ' -f /tmp/input-data'
 
-    calicoctl_bin = os.environ.get("CALICOCTL", "/code/bin/calicoctl-linux-amd64")
+    calicoctl_bin = os.environ.get("CALICOCTL", "/code/bin/calicoctl-linux-s390x")
 
     if ETCD_SCHEME == "https":
         etcd_auth = "%s:2379" % ETCD_HOSTNAME_SSL
