diff --git a/Makefile b/Makefile
index 1ad9669f..fe8934d6 100644
--- a/Makefile
+++ b/Makefile
@@ -501,17 +501,17 @@ $(RELEASE_DIR_BIN)/%:
 bin/helm: bin/helm3
 	mkdir -p bin
 	$(eval TMP := $(shell mktemp -d))
-	wget -q https://storage.googleapis.com/kubernetes-helm/helm-v2.16.3-linux-amd64.tar.gz -O $(TMP)/helm.tar.gz
+	wget -q https://storage.googleapis.com/kubernetes-helm/helm-v2.16.3-linux-s390x.tar.gz -O $(TMP)/helm.tar.gz
 	tar -zxvf $(TMP)/helm.tar.gz -C $(TMP)
-	mv $(TMP)/linux-amd64/helm bin/helm
+	mv $(TMP)/linux-s390x/helm bin/helm
 
 helm-deps: bin/helm3 bin/helm
 bin/helm3:
 	mkdir -p bin
 	$(eval TMP := $(shell mktemp -d))
-	wget -q https://get.helm.sh/helm-v3.3.1-linux-amd64.tar.gz -O $(TMP)/helm3.tar.gz
+	wget -q https://get.helm.sh/helm-v3.3.1-linux-s390x.tar.gz -O $(TMP)/helm3.tar.gz
 	tar -zxvf $(TMP)/helm3.tar.gz -C $(TMP)
-	mv $(TMP)/linux-amd64/helm bin/helm3
+	mv $(TMP)/linux-s390x/helm bin/helm3
 
 .PHONY: values.yaml
 values.yaml: _includes/charts/calico/values.yaml _includes/charts/tigera-operator/values.yaml
