--- go.mod	2020-12-09 08:42:25.037892000 +0000
+++ go_new.mod	2020-12-09 08:43:37.524910000 +0000
@@ -22,7 +22,7 @@
 	github.com/garyburd/redigo v0.0.0-20150301180006-535138d7bcd7
 	github.com/golang/snappy v0.0.0-20150730031844-723cc1e459b8 // indirect
 	github.com/influxdb/influxdb v0.9.6-0.20151125225445-9eab56311373
-	github.com/klauspost/crc32 v0.0.0-20151223135126-a3b15ae34567 // indirect
+	github.com/klauspost/crc32 v1.2.0 // indirect
 	github.com/mesos/mesos-go v0.0.7-0.20180413204204-29de6ff97b48
 	github.com/onsi/ginkgo v1.11.0 // indirect
 	github.com/onsi/gomega v1.7.1 // indirect
