diff --git a/upb/upb.c b/upb/upb.c
index 266ea7d7..39dddcbf 100644
--- a/upb/upb.c
+++ b/upb/upb.c
@@ -37,7 +37,7 @@ const char *upb_status_errmsg(const upb_status *status) { return status->msg; }
 void upb_status_seterrmsg(upb_status *status, const char *msg) {
   if (!status) return;
   status->ok = false;
-  strncpy(status->msg, msg, sizeof(status->msg));
+  strncpy(status->msg, msg, sizeof(status->msg) -1 );
   nullz(status);
 }
 

