diff --git a/tests/bugs/distribute/bug-1193636.c b/tests/bugs/distribute/bug-1193636.c
index ea3f79a4e0..e4b8c6e546 100644
--- a/tests/bugs/distribute/bug-1193636.c
+++ b/tests/bugs/distribute/bug-1193636.c
@@ -4,6 +4,11 @@
 #include <sys/xattr.h>
 #include <fcntl.h>
 #include <string.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
 
 #define MY_XATTR_NAME "user.ftest"
 #define MY_XATTR_VAL "ftestval"
diff --git a/tests/bugs/fuse/bug-1126048.c b/tests/bugs/fuse/bug-1126048.c
index 19165ecf6f..69f093f92f 100644
--- a/tests/bugs/fuse/bug-1126048.c
+++ b/tests/bugs/fuse/bug-1126048.c
@@ -5,6 +5,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <string.h>
+#include <stdlib.h>
 
 /*
  * This function opens a file and to trigger migration failure, unlinks the
diff --git a/tests/bugs/protocol/bug-808400-fcntl.c b/tests/bugs/protocol/bug-808400-fcntl.c
index 1b64636f18..7d646565c9 100644
--- a/tests/bugs/protocol/bug-808400-fcntl.c
+++ b/tests/bugs/protocol/bug-808400-fcntl.c
@@ -6,9 +6,15 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/wait.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
 
 #ifndef linux
-#define fstat64(fd, st) fstat(fd, st)
+#define fstat(fd, st) fstat(fd, st)
 #endif
 
 int
diff --git a/tests/bugs/protocol/bug-808400-flock.c b/tests/bugs/protocol/bug-808400-flock.c
index 54a507cc22..b544738ccd 100644
--- a/tests/bugs/protocol/bug-808400-flock.c
+++ b/tests/bugs/protocol/bug-808400-flock.c
@@ -6,9 +6,15 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/wait.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
 
 #ifndef linux
-#define fstat64(fd, st) fstat(fd, st)
+#define fstat(fd, st) fstat(fd, st)
 #endif
 
 int
diff --git a/tests/bugs/write-behind/bug-1058663.c b/tests/bugs/write-behind/bug-1058663.c
index aedf97d748..7f3f9f0158 100644
--- a/tests/bugs/write-behind/bug-1058663.c
+++ b/tests/bugs/write-behind/bug-1058663.c
@@ -5,6 +5,16 @@
 #include <sys/stat.h>
 #include <sys/mman.h>
 #include <signal.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
+
 
 #define FILE_SIZE 1048576
 
diff --git a/xlators/mgmt/glusterd/src/glusterd-utils.c b/xlators/mgmt/glusterd/src/glusterd-utils.c
index 436762802f..25a8188ea1 100644
--- a/xlators/mgmt/glusterd/src/glusterd-utils.c
+++ b/xlators/mgmt/glusterd/src/glusterd-utils.c
@@ -3232,7 +3232,7 @@ glusterd_dict_searialize(dict_t *dict_arr[], unsigned int count,
 {
     int i = 0;
     uint32_t keylen = 0;
-    uint64_t netword = 0;
+    uint32_t netword = 0;
     data_pair_t *pair = NULL;
     unsigned int dict_count = 0;
     int ret = 0;
