diff --git a/tests/bugs/shard/bug-1251824.t b/tests/bugs/shard/bug-1251824.t
index d81685d01d..75753d6910 100644
--- a/tests/bugs/shard/bug-1251824.t
+++ b/tests/bugs/shard/bug-1251824.t
@@ -14,7 +14,7 @@ TEST $CLI volume set $V0 features.shard-block-size 4MB
 TEST $CLI volume start $V0
 
 TEST glusterfs --volfile-id=$V0 --volfile-server=$H0 $M0
-TEST useradd -M test_user 2>/dev/null
+TEST useradd -UM test_user 2>/dev/null
 
 # Create 3 files as root.
 TEST touch $M0/foo
@@ -101,7 +101,7 @@ TEST run_cmd_as_user test_user "cp $M0/foo $M0/dir/quux"
 md5sum_foo=$(md5sum $M0/foo | awk '{print $1}')
 EXPECT "$md5sum_foo" echo `md5sum $M0/dir/quux | awk '{print $1}'`
 
-userdel test_user
+groupdel test_user
 
 TEST $CLI volume stop $V0
 TEST $CLI volume delete $V0
diff --git a/tests/bugs/shard/bug-1468483.t b/tests/bugs/shard/bug-1468483.t
index e462b8d54d..4419f61459 100644
--- a/tests/bugs/shard/bug-1468483.t
+++ b/tests/bugs/shard/bug-1468483.t
@@ -35,7 +35,7 @@ TEST stat $B0/${V0}0/.shard/$gfid_foo.6
 
 EXPECT "7" echo `$CLI volume profile $V0 info incremental | grep -w FSYNC | awk '{print $8}'`
 
-useradd -M test_user 2>/dev/null
+useradd -UM test_user 2>/dev/null
 
 TEST touch $M0/bar
 
@@ -51,7 +51,7 @@ TEST run_cmd_as_user test_user "dd if=/dev/zero conv=fsync of=$M0/bar bs=1M coun
 EXPECT "7" echo `$CLI volume profile $V0 info incremental | grep -w FSYNC | awk '{print $8}'`
 
 EXPECT_WITHIN $UMOUNT_TIMEOUT "Y" force_umount $M0
-userdel test_user
+groupdel test_user
 TEST $CLI volume stop $V0
 TEST $CLI volume delete $V0
 
