diff --git a/build/assets.sh b/build/assets.sh
index 7faf3300..19d7b425 100755
--- a/build/assets.sh
+++ b/build/assets.sh
@@ -30,7 +30,7 @@ FORCE="${FORCE:-}" # Force assets to be rebuilt if FORCE=true

 # Install while in a temp dir to avoid polluting go.mod/go.sum
 pushd "${TMPDIR:-/tmp}" > /dev/null
-go install github.com/kevinburke/go-bindata/go-bindata@latest
+go install github.com/kevinburke/go-bindata/go-bindata@v3.24.0
 popd > /dev/null

 build_asset () {
