diff --git a/CMakeLists.txt b/CMakeLists.txt
index 73e6df8..92b0b89 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -222,7 +222,7 @@ INCLUDE(FindCouchbasePythonInterp)
 INCLUDE(FindCouchbaseErlang)
 INCLUDE(FindCouchbaseDtrace)
 INCLUDE(FindCouchbaseGo)
-INCLUDE(FindCouchbaseBreakpad)
+# INCLUDE(FindCouchbaseBreakpad)
 INCLUDE(FindCouchbaseFlatbuffers)
 INCLUDE(FindCouchbaseValgrind)
 INCLUDE(FindCouchbaseV8)
diff --git a/cmake/Modules/CBDownloadDeps.cmake b/cmake/Modules/CBDownloadDeps.cmake
index 685e0f6..31ea3e3 100644
--- a/cmake/Modules/CBDownloadDeps.cmake
+++ b/cmake/Modules/CBDownloadDeps.cmake
@@ -308,6 +308,8 @@ IF (NOT CBDownloadDeps_INCLUDED)
       # Presumed Linux
       IF (_arch STREQUAL "aarch64")
         SET (_gofile "go${GOVERSION}.linux-arm64.tar.gz")
+    ELSEIF (_arch STREQUAL "s390x")
+        SET (_gofile "go${GOVERSION}.linux-s390x.tar.gz")
       ELSE ()
         SET (_gofile "go${GOVERSION}.linux-amd64.tar.gz")
       ENDIF ()
diff --git a/cmake/Modules/FindCouchbaseFolly.cmake b/cmake/Modules/FindCouchbaseFolly.cmake
index 9db03c5..5cda330 100644
--- a/cmake/Modules/FindCouchbaseFolly.cmake
+++ b/cmake/Modules/FindCouchbaseFolly.cmake
@@ -9,6 +9,7 @@ INCLUDE(FindCouchbaseDoubleConversion)
 INCLUDE(FindCouchbaseGlog)
 INCLUDE(FindCouchbaseLibevent)
 INCLUDE(FindCouchbaseOpenSSL)
+INCLUDE(FindCouchbaseGflags)
 
 include(PlatformIntrospection)
 include(SelectLibraryConfigurations)
@@ -88,6 +89,7 @@ endif()
 set(folly_dependancies ${DOUBLE_CONVERSION_LIBRARIES}
             ${GLOG_LIBRARIES}
             ${CMAKE_DL_LIBS}
+           ${GFLAGS_LIBRARIES}
             Boost::context
             Boost::filesystem
             Boost::program_options
@@ -134,7 +136,8 @@ target_include_directories(Folly::headers INTERFACE
         ${FOLLY_CONFIG_INCLUDE_DIR}
         ${Boost_INCLUDE_DIR}
         ${DOUBLE_CONVERSION_INCLUDE_DIR}
-        ${GLOG_INCLUDE_DIR})
+        ${GLOG_INCLUDE_DIR}
+       ${GFLAGS_INCLUDE_DIR})
 
 # Append Folly's depenancies to the include / lib variables so users
 # of Folly pickup the dependancies automatically.
@@ -144,6 +147,7 @@ set(FOLLY_INCLUDE_DIR
     ${Boost_INCLUDE_DIR}
     ${DOUBLE_CONVERSION_INCLUDE_DIR}
     ${GLOG_INCLUDE_DIR}
+    ${GFLAGS_INCLUDE_DIR}
     CACHE STRING "Folly include directories" FORCE)
 
 foreach(variant FOLLY_LIBRARIES FOLLY_LIBRARIES_UNSANITIZED)
diff --git a/deps/manifest.cmake b/deps/manifest.cmake
index 80ee1b0..1b1b2e2 100644
--- a/deps/manifest.cmake
+++ b/deps/manifest.cmake
@@ -17,7 +17,7 @@
 # in this same directory!
 #
 DECLARE_DEP (benchmark VERSION v1.6.0-cb1 PLATFORMS amzn2 centos7 debian9 debian10 macosx suse12 suse15 ubuntu18.04 ubuntu20.04 windows_msvc2017)
-DECLARE_DEP (breakpad V2 VERSION 20200430 BUILD 1 PLATFORMS amzn2 centos7 debian9 debian10 suse12 suse15 ubuntu18.04 ubuntu20.04 windows_msvc2017)
+# DECLARE_DEP (breakpad V2 VERSION 20200430 BUILD 1 PLATFORMS amzn2 centos7 debian9 debian10 suse12 suse15 ubuntu18.04 ubuntu20.04 windows_msvc2017)
 DECLARE_DEP (boost VERSION 1.74.0-cb1 PLATFORMS amzn2 centos7 debian9 debian10 macosx suse12 suse15 ubuntu18.04 ubuntu20.04 windows_msvc2017)
 DECLARE_DEP (cbpy VERSION 7.1.0-cb11 PLATFORMS linux macosx windows DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/python/interp")
 DECLARE_DEP (curl V2 VERSION 7.78.0 BUILD 7 PLATFORMS linux macosx windows)
diff --git a/cmake/Modules/FindCouchbaseGflags.cmake b/cmake/Modules/FindCouchbaseGflags.cmake
new file mode 100644
index 0000000..390fb8a
--- /dev/null
+++ b/cmake/Modules/FindCouchbaseGflags.cmake
@@ -0,0 +1,18 @@
+# Locate Google Glog library
+# This module defines
+#  GLOG_LIBRARIES, Library path and libs
+#  GLOG_INCLUDE_DIR, where to find the headers
+set(_gflags_exploded /usr/local)
+
+find_path(GFLAGS_INCLUDE_DIR gflags/gflags.h
+          PATH_SUFFIXES include
+          PATHS ${_gflags_exploded} /usr)
+
+find_library(GFLAGS_LIBRARIES
+             NAMES gflags
+             HINTS ${_gflags_exploded}/lib
+             /usr/lib/s390x-linux-gnu
+ .           /usr/lib64)
+message(STATUS "Found gflag at ${GFLAGS_INCLUDE_DIR} and ${GFLAGS_LIBRARIES}")
+
+mark_as_advanced(GFLAGS_INCLUDE_DIR GFLAGS_LIBRARIES)

diff --git a/cmake/Modules/PlatformIntrospection.cmake b/cmake/Modules/PlatformIntrospection.cmake
index 0207386..6f08c4e 100644
--- a/cmake/Modules/PlatformIntrospection.cmake
+++ b/cmake/Modules/PlatformIntrospection.cmake
@@ -246,6 +246,7 @@ MACRO (CB_GET_SUPPORTED_PLATFORM _is_supported_platform)
        "centos6" "centos7" "centos8"
        "debian8" "debian9" "debian10"
        "macosx"
+       "rhel7"
        "rhel8"
        "suse12" "suse15"
        "ubuntu16.04" "ubuntu18.04" "ubuntu20.04"
