diff --git a/third_party/aws/BUILD.bazel b/third_party/aws/BUILD.bazel
index fd355eeceb..4c8b7e7983 100644
--- a/third_party/aws/BUILD.bazel
+++ b/third_party/aws/BUILD.bazel
@@ -30,6 +30,9 @@ cc_library(
         "@org_tensorflow//tensorflow:freebsd": glob([
             "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",
         ]),
+        "@org_tensorflow//tensorflow:linux_s390x": glob([
+            "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",
+        ]),
         "//conditions:default": [],
     }) + glob([
         "aws-cpp-sdk-core/include/**/*.h",
