diff --git a/Makefile b/Makefile
index 7d9bbd5e26..701bfe399f 100644
--- a/Makefile
+++ b/Makefile
@@ -18,13 +18,13 @@ SUBDIRS := http ui chronograf query storage
 export GOOS=$(shell go env GOOS)
 export GOARCH=$(shell go env GOARCH)
 
-ifeq ($(GOARCH), amd64)
+ifneq (,$(filter $(GOARCH),amd64 s390x))
 	# Including the assets tag requires the UI to be built for compilation to succeed.
 	# Don't force it for running tests.
 	GO_TEST_TAGS :=
 	GO_BUILD_TAGS := assets
 else
-	# noasm needed to avoid a panic in Flux for non-amd64.
+	# noasm needed to avoid a panic in Flux for non-amd64,non-s390x.
 	GO_TEST_TAGS := noasm
 	GO_BUILD_TAGS := assets,noasm
 endif
@@ -228,7 +228,7 @@ docker-image-influx:
 docker-image-ui:
 	@cp .gitignore .dockerignore
 	@docker image build -t influxui:dev --target ui .
-	
+
 dshell-image:
 	@cp .gitignore .dockerignore
 	@docker image build --build-arg "USERID=$(shell id -u)" -t influxdb:dshell --target dshell .
diff --git a/go.mod b/go.mod
index 87f00e5189..8ea78a3288 100644
--- a/go.mod
+++ b/go.mod
@@ -127,3 +127,5 @@ require (
 replace github.com/apache/arrow/go/arrow v0.0.0-20191024131854-af6fa24be0db => github.com/influxdata/arrow/go/arrow v0.0.0-20200917142114-986e413c1705
 
 replace github.com/nats-io/nats-streaming-server v0.11.2 => github.com/influxdata/nats-streaming-server v0.11.3-0.20201112040610-c277f7560803
+
+replace github.com/apache/arrow/go/arrow v0.0.0-20200923215132-ac86123a3f01 => ../arrow/go/arrow
diff --git a/query/influxql/testdata/series_agg_7.out.json b/query/influxql/testdata/series_agg_7.out.json
index 4cf8d3b61b..9ac9bf0862 100644
--- a/query/influxql/testdata/series_agg_7.out.json
+++ b/query/influxql/testdata/series_agg_7.out.json
@@ -1 +1 @@
-{"results":[{"statement_id":0,"series":[{"name":"m","columns":["time","sum"],"values":[["1970-01-01T00:00:00Z",161.9342226725855]]}]}]}
\ No newline at end of file
+{"results":[{"statement_id":0,"series":[{"name":"m","columns":["time","sum"],"values":[["1970-01-01T00:00:00Z",161.93422267258558]]}]}]}
