diff --git a/docker-images/kafka-based/kafka/Makefile b/docker-images/kafka-based/kafka/Makefile
index d5fde4588..3f66f50f5 100644
--- a/docker-images/kafka-based/kafka/Makefile
+++ b/docker-images/kafka-based/kafka/Makefile
@@ -42,6 +42,8 @@ clean:
 .kafka-$(KAFKA_VERSION).tmp: ../../artifacts/binaries/kafka/archives/kafka_2.13-$(KAFKA_VERSION).tgz
 	test -d tmp/$(KAFKA_VERSION) || mkdir -p tmp/$(KAFKA_VERSION)
 	tar xvfz ../../artifacts/binaries/kafka/archives/kafka_2.13-$(KAFKA_VERSION).tgz -C tmp/$(KAFKA_VERSION) --strip-components=1
+	test -f tmp/$(KAFKA_VERSION)/libs/rocksdbjni-6.19.3.jar && cp -f /tmp/opertemp/libs/rocksdbjni-6.19.3.jar tmp/$(KAFKA_VERSION)/libs/ || true
+	test -f tmp/$(KAFKA_VERSION)/libs/rocksdbjni-6.22.1.1.jar && cp -f /tmp/opertemp/libs/rocksdbjni-6.22.1.1.jar tmp/$(KAFKA_VERSION)/libs/ || true
 	touch .kafka-$(KAFKA_VERSION).tmp
 
 .kafka-$(KAFKA_VERSION)-ignorelist.tmp: ../../artifacts/binaries/kafka/$(KAFKA_VERSION).ignorelist
diff --git a/docker-images/kaniko-executor/Makefile b/docker-images/kaniko-executor/Makefile
index 944ea5cd8..c9a47705a 100644
--- a/docker-images/kaniko-executor/Makefile
+++ b/docker-images/kaniko-executor/Makefile
@@ -5,7 +5,7 @@ docker_build:
 	# The Kaniko executor image used for building new Kafka Connect images with additional connectors is not build from
 	# scratch. We just pull the one released by Kaniko, retag it and push it to our repository to have our versioning on
 	# it and have it stored there in our other images.
-	$(DOCKER_CMD) pull $(DOCKER_PLATFORM) $(KANIKO_EXECUTOR)
+	# $(DOCKER_CMD) pull $(DOCKER_PLATFORM) $(KANIKO_EXECUTOR)
 	$(DOCKER_CMD) tag $(KANIKO_EXECUTOR) strimzi/$(PROJECT_NAME):latest
 	$(DOCKER_CMD) tag $(KANIKO_EXECUTOR) strimzi/$(PROJECT_NAME):$(BUILD_TAG)$(DOCKER_PLATFORM_TAG_SUFFIX)
 
