diff --git a/sql/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/ClientE2ETestSuite.scala b/sql/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/ClientE2ETestSuite.scala
index 3b6dd090..c99dde24 100644
--- a/sql/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/ClientE2ETestSuite.scala
+++ b/sql/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/ClientE2ETestSuite.scala
@@ -228,11 +228,12 @@ class ClientE2ETestSuite
   }

   test("spark deep recursion") {
+    var recursionDepth = if (System.getProperty("os.arch") == "s390x") 400 else 500
     var df = spark.range(1)
-    for (a <- 1 to 500) {
+    for (a <- 1 to recursionDepth) {
       df = df.union(spark.range(a, a + 1))
     }
-    assert(df.collect().length == 501)
+    assert(df.collect().length == recursionDepth + 1)
   }

   test("handle unknown exception") {
diff --git a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/UnsafeRowUtils.scala b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/UnsafeRowUtils.scala
index f2925314..8827d9fd 100644
--- a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/UnsafeRowUtils.scala
+++ b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/UnsafeRowUtils.scala
@@ -17,11 +17,15 @@

 package org.apache.spark.sql.catalyst.util

+import java.nio.ByteOrder.{nativeOrder, BIG_ENDIAN}
+
 import org.apache.spark.sql.catalyst.expressions.UnsafeRow
 import org.apache.spark.sql.types._

 object UnsafeRowUtils {

+  private final val isBigEndian = nativeOrder().equals(BIG_ENDIAN)
+
   /**
    * Use the following rules to check the integrity of the UnsafeRow:
    * - schema.fields.length == row.numFields should always be true
@@ -74,23 +78,23 @@ object UnsafeRowUtils {
       case (field, index) if UnsafeRow.isFixedLength(field.dataType) && !row.isNullAt(index) =>
         field.dataType match {
           case BooleanType =>
-            if ((row.getLong(index) >> 1) != 0L) {
+            if ((row.getByte(index) >> 1) != 0.toByte && getPadding(row, index, 1) != 0L) {
               return Some(s"Fixed-length field validation error: field: $field, index: $index")
             }
           case ByteType =>
-            if ((row.getLong(index) >> 8) != 0L) {
+            if (getPadding(row, index, 1) != 0L) {
               return Some(s"Fixed-length field validation error: field: $field, index: $index")
             }
           case ShortType =>
-            if ((row.getLong(index) >> 16) != 0L) {
+            if (getPadding(row, index, 2) != 0L) {
               return Some(s"Fixed-length field validation error: field: $field, index: $index")
             }
           case IntegerType =>
-            if ((row.getLong(index) >> 32) != 0L) {
+            if (getPadding(row, index, 4) != 0L) {
               return Some(s"Fixed-length field validation error: field: $field, index: $index")
             }
           case FloatType =>
-            if ((row.getLong(index) >> 32) != 0L) {
+            if (getPadding(row, index, 4) != 0L) {
               return Some(s"Fixed-length field validation error: field: $field, index: $index")
             }
           case _ =>
@@ -150,6 +154,14 @@ object UnsafeRowUtils {
     (offset, size)
   }

+  def getPadding(row: UnsafeRow, index: Int, valueLengthInBytes: Int): Long = {
+    val padding = row.getLong(index)
+    if (isBigEndian) {
+      return padding << (valueLengthInBytes * 8)
+    }
+    padding >> (valueLengthInBytes * 8)
+  }
+
   /**
    * Returns a Boolean indicating whether one should avoid calling
    * UnsafeRow.setNullAt for a field of the given data type.
diff --git a/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/FlatMapGroupsWithStateExecHelper.scala b/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/FlatMapGroupsWithStateExecHelper.scala
index b68c08b3..c2a690dc 100644
--- a/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/FlatMapGroupsWithStateExecHelper.scala
+++ b/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/FlatMapGroupsWithStateExecHelper.scala
@@ -152,7 +152,7 @@ object FlatMapGroupsWithStateExecHelper {
       shouldStoreTimestamp: Boolean) extends StateManagerImplBase(shouldStoreTimestamp) {

     private val timestampTimeoutAttribute =
-      AttributeReference("timeoutTimestamp", dataType = IntegerType, nullable = false)()
+      AttributeReference("timeoutTimestamp", dataType = LongType, nullable = false)()

     private val stateAttributes: Seq[Attribute] = {
       val encSchemaAttribs = toAttributes(stateEncoder.schema)
diff --git a/sql/core/src/test/scala/org/apache/spark/sql/execution/CoalesceShufflePartitionsSuite.scala b/sql/core/src/test/scala/org/apache/spark/sql/execution/CoalesceShufflePartitionsSuite.scala
index da43b0cf..01506c1f 100644
--- a/sql/core/src/test/scala/org/apache/spark/sql/execution/CoalesceShufflePartitionsSuite.scala
+++ b/sql/core/src/test/scala/org/apache/spark/sql/execution/CoalesceShufflePartitionsSuite.scala
@@ -17,6 +17,8 @@

 package org.apache.spark.sql.execution

+import java.nio.ByteOrder.{nativeOrder, BIG_ENDIAN}
+
 import org.apache.spark.{SparkConf, SparkFunSuite}
 import org.apache.spark.internal.config.IO_ENCRYPTION_ENABLED
 import org.apache.spark.internal.config.UI.UI_ENABLED
@@ -176,16 +178,21 @@ class CoalesceShufflePartitionsSuite extends SparkFunSuite with SQLConfHelper

     test(s"determining the number of reducers: complex query 1$testNameNote") {
       val test: (SparkSession) => Unit = { spark: SparkSession =>
+        // The default lz4 compression does not generate the same compressed
+        // stream on all architectures especially for different endianness.
+        // Increase the maxRange on big endian so the same number of partitions
+        // are generated as on little endian.
+        val maxRange = if (nativeOrder().equals(BIG_ENDIAN)) 2000 else 1000
         val df1 =
           spark
-            .range(0, 1000, 1, numInputPartitions)
+            .range(0, maxRange, 1, numInputPartitions)
             .selectExpr("id % 500 as key1", "id as value1")
             .groupBy("key1")
             .count()
             .toDF("key1", "cnt1")
         val df2 =
           spark
-            .range(0, 1000, 1, numInputPartitions)
+            .range(0, maxRange, 1, numInputPartitions)
             .selectExpr("id % 500 as key2", "id as value2")
             .groupBy("key2")
             .count()
@@ -197,7 +204,7 @@ class CoalesceShufflePartitionsSuite extends SparkFunSuite with SQLConfHelper
         val expectedAnswer =
           spark
             .range(0, 500)
-            .selectExpr("id", "2 as cnt")
+            .selectExpr("id", s"${maxRange/500}L as cnt")
         QueryTest.checkAnswer(
           join,
           expectedAnswer.collect().toImmutableArraySeq)
diff --git a/sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala b/sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala
index 272be70f..51762854 100644
--- a/sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala
+++ b/sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala
@@ -1127,7 +1127,7 @@ class AdaptiveQueryExecSuite
         SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1",
         SQLConf.SHUFFLE_PARTITIONS.key -> "100",
         SQLConf.SKEW_JOIN_SKEWED_PARTITION_THRESHOLD.key -> "800",
-        SQLConf.ADVISORY_PARTITION_SIZE_IN_BYTES.key -> "1000") {
+        SQLConf.ADVISORY_PARTITION_SIZE_IN_BYTES.key -> "900") {
         withTempView("skewData1", "skewData2") {
           spark
             .range(0, 1000, 1, 10)
diff --git a/sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/FlatMapGroupsWithStateExecHelperSuite.scala b/sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/FlatMapGroupsWithStateExecHelperSuite.scala
index 2456999b..63e9164a 100644
--- a/sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/FlatMapGroupsWithStateExecHelperSuite.scala
+++ b/sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/FlatMapGroupsWithStateExecHelperSuite.scala
@@ -42,7 +42,7 @@ class FlatMapGroupsWithStateExecHelperSuite extends StreamTest {
   test(s"StateManager v1 - primitive type - with timestamp") {
     val schema = new StructType()
       .add("value", IntegerType, nullable = false)
-      .add("timeoutTimestamp", IntegerType, nullable = false)
+      .add("timeoutTimestamp", LongType, nullable = false)
     testStateManagerWithTimestamp[Int](version = 1, schema, Seq(0, 10))
   }

@@ -75,7 +75,7 @@ class FlatMapGroupsWithStateExecHelperSuite extends StreamTest {
         StructField("d", DoubleType, nullable = false),
         StructField("str", StringType))
       )),
-      StructField("timeoutTimestamp", IntegerType, nullable = false)
+      StructField("timeoutTimestamp", LongType, nullable = false)
     ))

     val testValues = Seq(
