diff --git a/pkg/agent/plugin/nodeattestor/tpmdevid/devid_test.go b/pkg/agent/plugin/nodeattestor/tpmdevid/devid_test.go
index 0e4966ecb..c361769da 100644
--- a/pkg/agent/plugin/nodeattestor/tpmdevid/devid_test.go
+++ b/pkg/agent/plugin/nodeattestor/tpmdevid/devid_test.go
@@ -1,4 +1,4 @@
-//go:build !darwin
+//go:build !darwin && !s390x
 
 package tpmdevid_test
 
diff --git a/pkg/agent/plugin/nodeattestor/tpmdevid/tpmutil/session_test.go b/pkg/agent/plugin/nodeattestor/tpmdevid/tpmutil/session_test.go
index 3c2d485f7..f1c46f1b0 100644
--- a/pkg/agent/plugin/nodeattestor/tpmdevid/tpmutil/session_test.go
+++ b/pkg/agent/plugin/nodeattestor/tpmdevid/tpmutil/session_test.go
@@ -1,4 +1,4 @@
-//go:build !darwin
+//go:build !darwin && !s390x
 
 package tpmutil_test
 
diff --git a/pkg/server/plugin/nodeattestor/tpmdevid/devid_test.go b/pkg/server/plugin/nodeattestor/tpmdevid/devid_test.go
index 90603286c..69b817969 100644
--- a/pkg/server/plugin/nodeattestor/tpmdevid/devid_test.go
+++ b/pkg/server/plugin/nodeattestor/tpmdevid/devid_test.go
@@ -1,4 +1,4 @@
-//go:build !darwin
+//go:build !darwin && !s390x
 
 package tpmdevid_test
 
diff --git a/test/tpmsimulator/simulator.go b/test/tpmsimulator/simulator.go
index 7857f9376..b486a8e97 100644
--- a/test/tpmsimulator/simulator.go
+++ b/test/tpmsimulator/simulator.go
@@ -1,4 +1,4 @@
-//go:build !darwin
+//go:build !darwin && !s390x
 
 package tpmsimulator
 
