diff --git a/Dockerfile b/Dockerfile
index bb53bde..ddd9e0a 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -4,7 +4,7 @@
 ARG JAVA_MAJOR
 ARG DISTRIB_NAME=rockylinux
 ARG DISTRIB_MAJOR
-ARG IMAGE_JAVA_REPO=quay.io/alfresco
+ARG IMAGE_JAVA_REPO=alfresco
 ARG IMAGE_JAVA_NAME=alfresco-base-java
 ARG IMAGE_JAVA_TAG=jre${JAVA_MAJOR}-${DISTRIB_NAME}${DISTRIB_MAJOR}
 
@@ -95,39 +95,30 @@ ARG INSTALL_DIR=/usr/local
 COPY --from=tomcat_dist /build/tcnative $BUILD_DIR/tcnative
 COPY --from=tomcat_dist /build/apr $BUILD_DIR/apr
 SHELL ["/bin/bash", "-euo", "pipefail", "-c"]
-RUN <<EOT
-  yum install -y gcc make expat-devel java-${JAVA_MAJOR}-openjdk-devel redhat-rpm-config
-  yum clean all
-EOT
+RUN yum install -y gcc make expat-devel java-${JAVA_MAJOR}-openjdk-devel redhat-rpm-config && \
+    yum clean all
 
 WORKDIR ${BUILD_DIR}/apr
-RUN <<EOT
-  ./configure --prefix=${INSTALL_DIR}/apr
-  make -j "$(nproc)"
-  make install
-EOT
+RUN ./configure --prefix=${INSTALL_DIR}/apr && \
+    make -j "$(nproc)" && \
+    make install
 
 WORKDIR ${BUILD_DIR}/tcnative/native
-RUN <<EOT
-  if [ $DISTRIB_MAJOR -eq 8 ]; then
-    dnf install -y dnf-plugins-core
-    dnf config-manager -y --set-enabled powertools
-    dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
-    dnf install -y openssl3-devel
-    ln -s /usr/include/openssl3/openssl /usr/include/openssl
-    export LIBS="-L/usr/lib64/openssl3 -Wl,-rpath,/usr/lib64/openssl3 -lssl -lcrypto"
-    export CFLAGS="-I/usr/include/openssl3"
-    else dnf install -y openssl-devel
-  fi
-  dnf clean all
-  ./configure \
-    --libdir=${INSTALL_DIR}/tcnative \
-    --with-apr=${INSTALL_DIR}/apr/bin/apr-1-config \
-    --with-java-home="$JAVA_HOME" \
-    --disable-openssl-version-check
-  make -j "$(nproc)"
-  make install
-EOT
+RUN if [ "$DISTRIB_MAJOR" -eq 8 ]; then \
+      dnf install -y dnf-plugins-core && \
+      dnf config-manager -y --set-enabled powertools && \
+      dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && \
+      dnf install -y openssl3-devel && \
+      ln -s /usr/include/openssl3/openssl /usr/include/openssl && \
+      export LIBS="-L/usr/lib64/openssl3 -Wl,-rpath,/usr/lib64/openssl3 -lssl -lcrypto" && \
+      export CFLAGS="-I/usr/include/openssl3"; \
+    else \
+      dnf install -y openssl-devel; \
+    fi && \
+    dnf clean all && \
+    ./configure --libdir=${INSTALL_DIR}/tcnative --with-apr=${INSTALL_DIR}/apr/bin/apr-1-config --with-java-home="$JAVA_HOME" --disable-openssl-version-check && \
+    make -j "$(nproc)" && \
+    make install
 
 # hadolint ignore=DL3006
 FROM tcnative_build-${DISTRIB_NAME} AS tcnative_build
@@ -168,25 +159,23 @@ WORKDIR $CATALINA_HOME
 # https://github.com/docker-library/tomcat/issues/35
 RUN groupadd --system tomcat && \
   useradd -M -s /bin/false --home $CATALINA_HOME --system --gid tomcat tomcat
-COPY --chown=:tomcat --chmod=640 --from=tomcat_dist /build/tomcat $CATALINA_HOME
-COPY --chown=:tomcat --chmod=640 --from=tcnative_build /usr/local/tcnative $TOMCAT_NATIVE_LIBDIR
-COPY --chown=:tomcat --chmod=640 --from=tcnative_build /usr/local/apr $APR_LIBDIR
+COPY --from=tomcat_dist /build/tomcat $CATALINA_HOME
+COPY --from=tcnative_build /usr/local/tcnative $TOMCAT_NATIVE_LIBDIR
+COPY --from=tcnative_build /usr/local/apr $APR_LIBDIR
+RUN chown -R tomcat:tomcat $CATALINA_HOME $TOMCAT_NATIVE_LIBDIR $APR_LIBDIR && chmod -R 640 $CATALINA_HOME $TOMCAT_NATIVE_LIBDIR $APR_LIBDIR
 SHELL ["/bin/bash", "-euo", "pipefail", "-c"]
-RUN <<EOT
-  if [ $DISTRIB_MAJOR -eq 8 ]; then
-    dnf install -y dnf-plugins-core
-    dnf config-manager -y --set-enabled powertools
-    dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
-    dnf install -y openssl3-libs
-    dnf clean all
-  fi
-  mkdir -m 770 logs temp work && chgrp tomcat . logs temp work
-  chmod ug+x bin/*.sh
-  find . -type d -exec chmod 770 {} +
-  # verify Tomcat Native is working properly
-  nativeLines="$(catalina.sh configtest 2>&1 | grep -c 'Loaded Apache Tomcat Native library')"
-  test $nativeLines -ge 1 || exit 1
-EOT
+RUN if [ "$DISTRIB_MAJOR" -eq 8 ]; then \
+      dnf install -y dnf-plugins-core && \
+      dnf config-manager -y --set-enabled powertools && \
+      dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && \
+      dnf install -y openssl3-libs && \
+      dnf clean all; \
+    fi && \
+    mkdir -m 770 logs temp work && chgrp tomcat . logs temp work && \
+    chmod ug+x bin/*.sh && \
+    find . -type d -exec chmod 770 {} + && \
+    nativeLines="$(catalina.sh configtest 2>&1 | grep -c 'Loaded Apache Tomcat Native library')" && \
+    test "$nativeLines" -ge 1
 
 USER tomcat
 EXPOSE 8080
