diff --git a/test/types.c b/test/types.c
index 9880296..8d643ba 100644
--- a/test/types.c
+++ b/test/types.c
@@ -297,7 +297,7 @@ ODBC_TEST(t_nobigint)
   SQLLEN nlen= 0;
   SQLSMALLINT type= 0;
   SQLULEN     size= 0;
-  SQLCHAR* name[4];
+  SQLCHAR name[4];
 
   OK_SIMPLE_STMT(Stmt, "DROP TABLE IF EXISTS t_nobigint");
 
@@ -324,7 +324,14 @@ ODBC_TEST(t_nobigint)
   is_num(size, 4);
   CHECK_STMT_RC(hstmt, SQLGetData(hstmt, 1, SQL_C_DEFAULT, &id, sizeof(id), &nlen));
 
-  is_num(0xFFFFFFFF0000000F, id);
+  if (little_endian() )
+  {
+    is_num(0xFFFFFFFF0000000F, id);
+  }
+  else
+  {
+    is_num(0x0000000FFFFFFFFF, id);
+  }
 
   CHECK_STMT_RC(hstmt, SQLFreeStmt(hstmt, SQL_CLOSE));

