--- original/Dockerfile.s390x.calico_test	2019-02-11 06:32:00.224757910 -0500
+++ updated/Dockerfile.s390x.calico_test	2019-02-04 05:25:48.324813968 -0500
@@ -46,7 +46,7 @@
 # Running STs in this container requires that it has all dependencies installed
 # for executing the tests. Install these dependencies:
 RUN apk add --update python python-dev py2-pip py-setuptools openssl-dev libffi-dev tshark \
-        netcat-openbsd iptables ip6tables iproute2 iputils ipset curl && \
+        netcat-openbsd iptables ip6tables iproute2 iputils ipset curl build-base && \
         echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf && \
         rm -rf /var/cache/apk/*
 
@@ -59,7 +59,7 @@
 
 # Install etcdctl
 COPY pkg /pkg/
-RUN tar -xzf pkg/etcd-v3.3.1-linux-s390x.tar.gz -C /usr/local/bin/
+RUN tar -xzf pkg/etcd-v3.3.7-linux-s390x.tar.gz -C /usr/local/bin/
 
 # The container is used by mounting the code-under-test to /code
 WORKDIR /code/
