--- a/bin/init.sh
+++ b/bin/init.sh
@@ -96,13 +96,15 @@ function download_envoy_if_necessary () {

     # Download and extract the binary to the output directory.
     echo "Downloading Envoy: ${DOWNLOAD_COMMAND} $1 to $2"
-    time ${DOWNLOAD_COMMAND} "$1" | tar xz

     # Copy the extracted binary to the output location
-    cp usr/local/bin/envoy "$2"
+    if [[ "$2" ==  "$ISTIO_ENVOY_LINUX_DEBUG_PATH" ]] ; then
+      cp <path-to-envoy-debug-binary/envoy> "$2"
+    else
+      cp <path-to-envoy-release-binary/envoy> "$2"
+    fi

     # Remove the extracted binary.
-    rm -rf usr

     # Make a copy named just "envoy" in the same directory (overwrite if necessary).
     echo "Copying $2 to $(dirname "$2")/envoy"
