--- Lib/test/test_zlib.py       2022-03-16 11:27:11.000000000 +0000
+++ tmp/Lib/test/test_zlib.py   2022-03-24 18:12:34.837420527 +0000
@@ -3,6 +3,7 @@
 from test.support import import_helper
 import binascii
 import copy
+import os
 import pickle
 import random
 import sys
@@ -18,6 +19,7 @@
         hasattr(zlib.decompressobj(), "copy"),
         'requires Decompress.copy()')

+skip_on_s390x = unittest.skipIf(hasattr(os, 'uname') and os.uname().machine == 's390x', 'skipped on s390x')

 class VersionTestCase(unittest.TestCase):

@@ -181,7 +183,8 @@
                                          wbits=zlib.MAX_WBITS,
                                          bufsize=zlib.DEF_BUF_SIZE),
                          HAMLET_SCENE)
-
+
+    @skip_on_s390x
     def test_speech128(self):
         # compress more data
         data = HAMLET_SCENE * 128
@@ -233,6 +236,7 @@

 class CompressObjectTestCase(BaseCompressTestCase, unittest.TestCase):
     # Test compression object
+    @skip_on_s390x
     def test_pair(self):
         # straightforward compress/decompress objects
         datasrc = HAMLET_SCENE * 128
