--- build_old.sh        2019-01-31 10:49:56.059810484 +0000
+++ build.sh    2019-01-24 05:24:56.594663607 +0000
@@ -19,4 +19,6 @@

 npm install
 npm install pkg@4.3.4
-node_modules/pkg/lib-es5/bin.js -t node6-linux-x64 .
+patch --ignore-whitespace node_modules/pkg-fetch/lib-es5/build.js  < pkg-fetch_build.diff
+patch --ignore-whitespace node_modules/pkg-fetch/lib-es5/system.js < pkg-fetch_system.diff
+node_modules/pkg/lib-es5/bin.js -t node6-linux-s390x .
