diff --git a/cmake/modules/luajit.cmake b/cmake/modules/luajit.cmake
index 4bd7873..0fee83e 100644
--- a/cmake/modules/luajit.cmake
+++ b/cmake/modules/luajit.cmake
@@ -18,8 +18,8 @@ if(NOT LUAJIT_INCLUDE)
   set(LUAJIT_LIB "${LUAJIT_SRC}/libluajit.a")
   externalproject_add(
     luajit
-    GIT_REPOSITORY "https://github.com/LuaJIT/LuaJIT"
-    GIT_TAG "1d8b747c161db457e032a023ebbff511f5de5ec2"
+    URL "https://github.com/linux-on-ibm-z/LuaJIT/archive/v2.1.zip"
+    PATCH_COMMAND patch -l "${PROJECT_BINARY_DIR}/luajit-prefix/src/luajit/src/lauxlib.h" $ENV{SOURCE_ROOT}/falco/lauxlib.h.patch
     CONFIGURE_COMMAND ""
     BUILD_COMMAND ${CMD_MAKE}
     BUILD_IN_SOURCE 1
