--- BUILD	2018-10-19 07:08:31.713977988 +0000
+++ bl	2018-10-19 08:40:47.914092827 +0000
@@ -43,7 +43,7 @@
         ":windows_x86_64": ["thirdparty_build/lib/luajit.lib"],
         "//conditions:default": ["thirdparty_build/lib/libluajit-5.1.a"],
     }),
-    hdrs = glob(["thirdparty_build/include/luajit-2.0/*"]),
+    hdrs = glob(["thirdparty_build/include/luajit-2.1/*"]),
     includes = ["thirdparty_build/include"],
     # TODO(mattklein123): We should strip luajit-2.0 here for consumers. However, if we do that
     # the headers get included using -I vs. -isystem which then causes old-style-cast warnings.
