--- a/bazel/repositories.bzl
+++ b/bazel/repositories.bzl
@@ -610,7 +610,7 @@ def _com_github_gperftools_gperftools():
     http_archive(
         name = "com_github_gperftools_gperftools",
         build_file_content = BUILD_ALL_CONTENT,
-        patch_cmds = ["./autogen.sh"],
+        #patch_cmds = ["./autogen.sh"],
         **location
     )

@@ -632,5 +632,8 @@ def _is_arch(ctxt, arch):
 def _is_linux_ppc(ctxt):
     return _is_linux(ctxt) and _is_arch(ctxt, "ppc")

+def _is_linux_s390x(ctxt):
+    return _is_linux(ctxt) and _is_arch(ctxt, "s390x")
+
 def _is_linux_x86_64(ctxt):
     return _is_linux(ctxt) and _is_arch(ctxt, "x86_64")
