diff --git a/lib/Makefile b/lib/Makefile
index 7c6dff02..a2deff07 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -220,13 +220,11 @@ libzstd : $(LIBZSTD)
 .PHONY: lib
 lib : libzstd.a libzstd
 
-.PHONY: lib-mt
 %-mt : CPPFLAGS += -DZSTD_MULTITHREAD
 %-mt : LDFLAGS  += -pthread
 %-mt : %
 	@echo multi-threading build completed
 
-.PHONY: lib-release
 %-release : DEBUGFLAGS :=
 %-release : %
 	@echo release build completed