diff --git a/include/flatbuffers/base.h b/include/flatbuffers/base.h
index c755d6f8..dcb33850 100644
--- a/include/flatbuffers/base.h
+++ b/include/flatbuffers/base.h
@@ -270,6 +270,11 @@ template<typename T> void WriteScalar(void *p, T t) {
   *reinterpret_cast<T *>(p) = EndianScalar(t);
 }

+template<typename T> struct Offset;
+template<typename T> void WriteScalar(void *p, Offset<T> t) {
+  *reinterpret_cast<uoffset_t *>(p) = EndianScalar(t.o);
+}
+
 // Computes how many bytes you'd have to pad to be able to write an
 // "scalar_size" scalar if the buffer had grown to "buf_size" (downwards in
 // memory).
 
